---
author: "Th√©o Lassale"
editor_options: 
  chunk_output_type: console
date: '`r Sys.Date()`'
output:
  html_document:
  theme: flatly
toc: true
toc_float: true
code_folding: hide
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```

```{r loaddata}
rm(list = ls())
```

```{r load_libraries}
library(tidyverse)
library(rtracklayer)
library(edgeR)
library(ggvenn)
library(ComplexHeatmap)
library(clusterProfiler)
library(DOSE)
library(enrichplot)
library(org.Mm.eg.db)
library(forcats)
library(reactome.db)
library(ReactomePA)
library(KEGGREST)
library(msigdb)
library(biomaRt)
library(immunedeconv)
library(gridExtra)
```


```{r annotation}
gtf <- import("/data/annotations/pipelines/Mouse/mm10/gtf/gencode.vM22.annotation.gtf")
gtf_annot <- gtf %>% as_tibble() %>% filter(type == "gene") %>%  mutate(gene_id = str_remove(gene_id, "\\..*"))


in.dir <- "Bulk_RNA_in_vivo/preprocessing_in_vivo_RNA/results_pipeline/"
annot_gene <- read_delim(file.path(in.dir,"counts/tableannot.csv"), delim = ",") %>%
  dplyr::select(gene_id, gene_name) %>%
  full_join(gtf_annot[, c("gene_id","gene_type")], by="gene_id")

```

```{r samplePlan}
sample_plan <- read.csv("Bulk_RNA_in_vivo/preprocessing_in_vivo_RNA/results_pipeline/samples/sample_plan.csv", header=FALSE)
colnames(sample_plan) <- c("sample ID", "sample Name", "Path_to_R1", "Path_to_R2")
rownames(sample_plan) <- NULL
sample_plan[grepl("Ct", sample_plan$`sample ID`) | grepl("CT", sample_plan$`sample ID`), "treatment"] <- "PBS"
sample_plan[1:3, "treatment"] <- "MTX"
sample_plan[4:6, "treatment"] <- "PBS"
sample_plan[!grepl("PBS", sample_plan$`sample ID`) & grepl("MTX", sample_plan$`sample ID`), "treatment"] <- "MTX"
sample_plan[grepl("Cis", sample_plan$`sample ID`),"treatment"] <- "Cis"
sample_plan[grepl("Oxa", sample_plan$`sample ID`),"treatment"] <- "Oxa"
sample_plan[grepl("D3", sample_plan$`sample ID`), "Time_point"] <- 3
sample_plan[grepl("D10", sample_plan$`sample ID`), "Time_point"] <- 10
sample_plan[grepl("D22", sample_plan$`sample ID`) | grepl("D24", sample_plan$`sample ID`) , "Time_point"] <- 22

sample_annot <- read.csv("Bulk_RNA_in_vivo/preprocessing_in_vivo_RNA/results_pipeline/samples/sample_annot.csv", check.names = FALSE)
sample_annot$`Sample ID` <- make.names(sample_annot$`Sample ID`)
sample_annot <- sample_annot %>% dplyr::rename("sample ID" = "Sample ID")
sample_plan$`sample ID`<- make.names(sample_plan$`sample ID`)
all_sample <- full_join(sample_plan, sample_annot, by = "sample ID")
all_sample$ID <- stringr::str_remove(stringr::str_remove_all(all_sample$`sample ID`, "_D3|_D10|_D22|.D22"), "MTX_PBS_|OCP_")

sample_plan <- tidyr::unite(all_sample, col = "new_name", Treatment, Day, Sheet, ID, remove = FALSE, sep = "_")
rownames(sample_plan) <- sample_plan$new_name
```



```{r counts_table}
counts <- read_delim(file.path(in.dir,"counts/tablecounts_raw.csv"), delim = ",") %>%
  dplyr::rename("gene_id" = "...1") %>% 
  inner_join(., annot_gene, by = "gene_id") %>% 
  relocate(gene_name, .after = gene_id) %>%
  relocate(gene_type, .after = gene_name)

colnames(counts) <- make.names(colnames(counts))
counts <- counts[,c("gene_id","gene_name", "gene_type", sample_plan$`sample ID`)]
colnames(counts)[4:ncol(counts)] <- sample_plan$new_name
```




# **3. Differential Expression Analysis** 

```{r data_prep}
# 1st dataset (counts_table + sample_plan) - MTX vs PBS
counts_MP <- counts %>% 
  dplyr::select(gene_id, gene_name, gene_type, contains("MTX") & !contains("CT1_D24")) %>% 
  as.data.frame() %>% 
  column_to_rownames("gene_id")

sample_plan_MP <- sample_plan %>% filter(new_name %in% colnames(counts_MP))
sample_plan_MP[,"Treatment"] <- paste(sample_plan_MP$treatment, sample_plan_MP$Day, sep ="_")


# 2nd dataset (counts_table + sample_plan) - Oxa/Cis vs PBS                          
counts_OCP <- counts %>% 
  dplyr::select(gene_id, gene_name, gene_type, contains("OCP") & !contains("CT1_D24")) %>% 
  as.data.frame() %>% 
  column_to_rownames("gene_id")

sample_plan_OCP <- sample_plan %>% filter(new_name %in% colnames(counts_OCP))
sample_plan_OCP[,"Treatment"] <- paste(sample_plan_OCP$treatment, sample_plan_OCP$Day, sep ="_")
```



```{r run_dea}
y_raw_MP <- DGEList(counts = dplyr::select(counts_MP, -c("gene_name","gene_type")) , samples = sample_plan_MP, genes = counts_MP$gene_name)
design_MP <- model.matrix(~ 0 + Treatment, data=y_raw_MP$samples)
colnames(design_MP) <- sub("Treatment","", colnames(design_MP))
keep <- filterByExpr(y_raw_MP,design_MP)
y_MP <- y_raw_MP[keep,]
y_MP <- calcNormFactors(y_MP, method="TMM")
v_MP <- voomWithQualityWeights(y_MP, design_MP)
fit_MP <- lmFit(v_MP, design_MP)


y_raw_OCP <- DGEList(counts = dplyr::select(counts_OCP, -c("gene_name","gene_type")) , samples = sample_plan_OCP, genes = counts_OCP$gene_name)
design_OCP <- model.matrix(~ 0 + Treatment, data=y_raw_OCP$samples)
colnames(design_OCP) <- sub("Treatment","", colnames(design_OCP))
keep <- filterByExpr(y_raw_OCP,design_OCP)
y_OCP <- y_raw_OCP[keep,]
y_OCP <- calcNormFactors(y_OCP, method="TMM")
v_OCP <- voomWithQualityWeights(y_OCP, design_OCP)
fit_OCP <- lmFit(v_OCP, design_OCP)
```


```{r get_contrasts}
contrast_MP <- makeContrasts(MP_D3 = MTX_D3 - PBS_D3, MP_D10 = MTX_D10 - PBS_D10, MP_D22 = MTX_D22 - PBS_D22, levels = design_MP)
fit_contrast_MP <- contrasts.fit(fit_MP, contrast_MP)
fit_contrast_MP <- eBayes(fit_contrast_MP)

contrast_OP <- makeContrasts(OP_D3 = Oxa_D3 - PBS_D3, OP_D10 = Oxa_D10 - PBS_D10, OP_D22 = Oxa_D22 - PBS_D22, levels = design_OCP)
fit_contrast_OP <- contrasts.fit(fit_OCP, contrast_OP)
fit_contrast_OP <- eBayes(fit_contrast_OP)


contrast_CP <- makeContrasts(CP_D3 = Cis_D3 - PBS_D3, CP_D10 = Cis_D10 - PBS_D10, CP_D22 = Cis_D22 - PBS_D22, levels = design_OCP)
fit_contrast_CP <- contrasts.fit(fit_OCP, contrast_CP)
fit_contrast_CP <- eBayes(fit_contrast_CP)

contrast_MM <- makeContrasts(MTX_D10_D3 = MTX_D10 - MTX_D3, MTX_D22_D10 = MTX_D22 - MTX_D10, MTX_D22_D3 = MTX_D22 - MTX_D3, levels = design_MP)
fit_contrast_MM <- contrasts.fit(fit_MP, contrast_MM)
fit_contrast_MM <- eBayes(fit_contrast_MM)

contrast_OO <- makeContrasts(Oxa_D10_D3 = Oxa_D10 - Oxa_D3, Oxa_D22_D10 = Oxa_D22 - Oxa_D10, Oxa_D22_D3 = Oxa_D22 - Oxa_D3, levels = design_OCP)
fit_contrast_OO <- contrasts.fit(fit_OCP, contrast_OO)
fit_contrast_OO <- eBayes(fit_contrast_OO)

contrast_CC <- makeContrasts(Cis_D10_D3 = Cis_D10 - Cis_D3, Cis_D22_D10 = Cis_D22 - Cis_D10, Cis_D22_D3 = Cis_D22 - Cis_D3, levels = design_OCP)
fit_contrast_CC <- contrasts.fit(fit_OCP, contrast_CC)
fit_contrast_CC <- eBayes(fit_contrast_CC)

contrast_PP_MTX <- makeContrasts(PBS_D10_D3 = PBS_D10 - PBS_D3, PBS_D22_D10 = PBS_D22 - PBS_D10, PBS_D22_D3 = PBS_D22 - PBS_D3, levels = design_OCP)
fit_contrast_PP_MTX <- contrasts.fit(fit_OCP, contrast_PP_MTX)
fit_contrast_PP_MTX <- eBayes(fit_contrast_PP_MTX)

contrast_PP_OCP <- makeContrasts(PBS_D10_D3 = PBS_D10 - PBS_D3, PBS_D22_D10 = PBS_D22 - PBS_D10, PBS_D22_D3 = PBS_D22 - PBS_D3, levels = design_OCP)
fit_contrast_PP_OCP <- contrasts.fit(fit_OCP, contrast_PP_OCP)
fit_contrast_PP_OCP <- eBayes(fit_contrast_PP_OCP)
```


```{r}
top_table_MP_D3 <- topTable(fit_contrast_MP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "MP_D3") %>% dplyr::mutate(contrast = "MTX_D3_vs_PBS_D3")
top_table_MP_D10 <- topTable(fit_contrast_MP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "MP_D10") %>% dplyr::mutate(contrast = "MTX_D10_vs_PBS_D10")
top_table_MP_D22 <- topTable(fit_contrast_MP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "MP_D22") %>% dplyr::mutate(contrast = "MTX_D22_vs_PBS_D22")

top_table_OP_D3 <- topTable(fit_contrast_OP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "OP_D3") %>% dplyr::mutate(contrast = "OXA_D3_vs_PBS_D3")
top_table_OP_D10 <- topTable(fit_contrast_OP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "OP_D10") %>% dplyr::mutate(contrast = "OXA_D10_vs_PBS_D10")
top_table_OP_D22 <- topTable(fit_contrast_OP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "OP_D22") %>% dplyr::mutate(contrast = "OXA_D22_vs_PBS_D22")

top_table_CP_D3 <- topTable(fit_contrast_CP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "CP_D3") %>% dplyr::mutate(contrast = "CIS_D3_vs_PBS_D3")
top_table_CP_D10 <- topTable(fit_contrast_CP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "CP_D10") %>% dplyr::mutate(contrast = "CIS_D10_vs_PBS_D10")
top_table_CP_D22 <- topTable(fit_contrast_CP, sort.by= "P", adjust.method = "BH", number = Inf, coef = "CP_D22") %>% dplyr::mutate(contrast = "CIS_D22_vs_PBS_D22")



top_table_final_in_vivo_MTX_OXA_CIS_Control <- rbind(top_table_MP_D3, top_table_MP_D10, top_table_MP_D22, top_table_OP_D3, top_table_OP_D10, top_table_OP_D22, top_table_CP_D3, top_table_CP_D10, top_table_CP_D22) %>% rownames_to_column("gene_id") %>% rename("gene_name" = "genes") %>% inner_join(counts_MP[,c("gene_name","gene_type")], by="gene_name", relationship ="many-to-many") %>% dplyr::relocate(gene_type, .before = "gene_name") %>% relocate(gene_name, .before = "gene_id")
write.csv(top_table_final_in_vivo_MTX_OXA_CIS_Control, "/data/users/tlassale/MODICED_link/top_table_in_vivo_for_Andrea.csv")

```




```{r}
get_results_dea <- function(fit_contrast, a, diff_gene_list, up_reg_genes_list, down_reg_genes_list, label, volcano_plot_colored_by_logFC_png, volcano_plot_colored_by_coding_genes_png, volcano_plot_colored_by_non_coding_genes_png, volcano_plot_colored_by_logFC_html, volcano_plot_colored_by_coding_genes_html, volcano_plot_colored_by_non_coding_genes_html, proportion_diff_gene_by_type, proportion_diff_gene_by_type_barplot, proportion_of_up_down_reg_by_type){
  
  # Top table (+ update Top table)
  genes <- topTable(fit_contrast, sort.by= "P", adjust.method = "BH", number = Inf, coef = a) %>%
  rownames_to_column("gene_id") %>%
  left_join(annot_gene[,c("gene_id","gene_type")], by="gene_id") %>%
  mutate(gene_type_summary = case_when(gene_type != "protein_coding" & gene_type != "Mt_tRNA" & gene_type != "Mt_rRNA" &  adj.P.Val < 0.05 & abs(logFC) > 1 ~ "significant non_coding_gene (not mitochondrial)", (gene_type == "Mt_tRNA" | gene_type == "Mt_rRNA") &  adj.P.Val < 0.05 & abs(logFC) > 1 ~ "significant non_coding_gene (mitochondrial)", gene_type == "protein_coding" & adj.P.Val < 0.05 & abs(logFC) > 1 ~ "significant protein_coding_gene", TRUE ~ "not_significant"))
  
  attach(genes)
  genes$gene_type_summary[which(gene_type_summary == "not_significant" & gene_type == "protein_coding")]  <- "not_significant protein_coding_gene"
  genes$gene_type_summary[which(gene_type_summary == "not_significant" & (gene_type == "Mt_tRNA" | gene_type == "Mt_rRNA"))]  <- "not_significant non_coding_gene (mitochondrial)"
  genes$gene_type_summary[which(gene_type_summary == "not_significant" & gene_type != "Mt_tRNA"
                                      & gene_type != "Mt_rRNA" & gene_type != "protein_coding")]  <- "not_significant non_coding_gene (not mitochondrial)"
  
  # Filter by adj p-value < 0.05
  diff_genes <- genes  %>%
    filter(adj.P.Val < 0.05, abs(logFC) > 1) %>% 
    dplyr::select(gene_id, genes, logFC, adj.P.Val, gene_type, gene_type_summary) %>% 
    mutate(group = if_else(logFC < 0, "down", "up"))
  
  #Concatenate in vectors (list of genes)
  diff_genes %>% 
    dplyr::select(genes, group) %>% 
    write.csv(diff_gene_list)
  
  # Concatenate in vectors (up reg genes)
  up_diff_genes <- diff_genes %>% dplyr::select(genes, group) %>%  
    filter(group == "up")  %>% 
    write.csv(file= up_reg_genes_list)
  
  # Concatenate in vectors (down reg genes)
  down_diff_genes <- diff_genes %>% dplyr::select(genes, group) %>%  
    filter(group == "down")  %>% 
    write.csv(file= down_reg_genes_list)
  
  
  # Volcano Plot - Colored by logFC/adj p-value
  volcano_by_logFC <- genes %>% 
    ggplot(aes(x=logFC, y=-log10(adj.P.Val), label=genes)) +
    geom_point(color="grey") +
    geom_point(data = filter(genes, logFC > 1 & adj.P.Val < 0.05), color="red") +
    geom_point(data = filter(genes, logFC < -1 & adj.P.Val < 0.05), color="blue") +
    geom_point(data = filter(genes, adj.P.Val < 0.05 & (logFC > 0 & logFC < 1) ), color="orange") +
    geom_point(data = filter(genes, adj.P.Val < 0.05 & (logFC < 0 & logFC >  -1)), color="lightblue") +
    geom_vline(xintercept = c(-1,1), colour = "black", linetype = "dashed") +
    geom_hline(yintercept = (-log10(0.05)), colour = "black", linetype = "dashed") +
    ylab("-log10(adj.P.value)") +
    xlab("log2(FoldChange)") +
    theme_linedraw() +
    ggtitle(label) +
    theme(plot.title=element_text(hjust=0.5, face="bold"))
  
  ggsave(volcano_plot_colored_by_logFC_png, width = 9, height = 5)



# Volcano plot -  Colored by coding genes
volcano_by_coding <- genes %>% subset(gene_type_summary == "significant protein_coding_gene" | gene_type_summary == "not_significant protein_coding_gene") %>% 
  ggplot(aes(x=logFC, y=-log10(adj.P.Val), label=genes)) +
  geom_point(mapping = aes(color=gene_type_summary), pch = 1, size = 3) +
  scale_color_manual(values = c("blue","red")) +
  geom_vline(xintercept = c(-1,1), colour = "black", linetype = "dashed") +
  geom_hline(yintercept = (-log10(0.05)), colour = "black", linetype = "dashed") +
  ylab("-log10(adj.P.value)") +
  xlab("log2(FoldChange)") +
  theme_linedraw() +
  ggtitle(label) +
  theme(plot.title=element_text(hjust=0.5, face="bold"))

ggsave(volcano_plot_colored_by_coding_genes_png, width=9, height=5)



# Volcano plot -  Colored by non-coding genes
mito_genes <- genes %>% filter(gene_type_summary == "significant non_coding_gene (mitochondrial)" | gene_type_summary == "not_significant non_coding_gene (mitochondrial)")
volcano_by_non_coding <- genes %>% subset(gene_type_summary != "significant protein_coding_gene" & gene_type_summary != "not_significant protein_coding_gene") %>% bind_rows(mito_genes) %>% 
  ggplot(aes(x=logFC, y=-log10(adj.P.Val), label=genes)) +
  geom_point(mapping = aes(color=gene_type_summary), pch = 1, size = 3) +
  scale_color_manual(values = c("blue","yellow", "green", "red")) +
  geom_vline(xintercept = c(-1,1), colour = "black", linetype = "dashed") +
  geom_hline(yintercept = (-log10(0.05)), colour = "black", linetype = "dashed") +
  ylab("-log10(adj.P.value)") +
  xlab("log2(FoldChange)") +
  theme_linedraw() +
  ggtitle(label) +
  theme(plot.title=element_text(hjust=0.5, face="bold"))


ggsave(volcano_plot_colored_by_non_coding_genes_png, width=9, height=5)


# Plotly + Save object
volcano_by_logFC_plotly <- plotly::ggplotly(volcano_by_logFC)
htmlwidgets::saveWidget(volcano_by_logFC_plotly, volcano_plot_colored_by_logFC_html)

volcano_by_coding_plotly <- plotly::ggplotly(volcano_by_coding)
htmlwidgets::saveWidget(volcano_by_coding_plotly, volcano_plot_colored_by_coding_genes_html)

volcano_by_non_coding_plotly <- plotly::ggplotly(volcano_by_non_coding)
htmlwidgets::saveWidget(volcano_by_non_coding_plotly, volcano_plot_colored_by_non_coding_genes_html)


# Get proportion of differentially expressed coding and non-coding genes
prop_by_type_diff_gene <- diff_genes %>%
  dplyr::group_by(gene_type_summary) %>%
  dplyr::summarise(count = n()) %>%
  mutate(prop = count / sum(count))
  
write.csv(prop_by_type_diff_gene, file=proportion_diff_gene_by_type, quote=FALSE, col.names = TRUE, row.names = FALSE)

# Corresponding Barplot
prop_by_type_diff_gene_barplot <- ggplot(prop_by_type_diff_gene, aes(x=gene_type_summary, y=prop)) + geom_bar(stat = "identity")
ggsave(filename = proportion_diff_gene_by_type_barplot)


# Get proportion of up and down regulated genes by gene type
prop_by_type_up_down_genes <- diff_genes %>%
  dplyr::count(gene_type_summary, group) %>%
  dplyr::group_by(group) %>%
  dplyr::mutate(prop = n / sum(n)) %>%
  ungroup()

write.csv(prop_by_type_up_down_genes, file=proportion_of_up_down_reg_by_type, quote=FALSE, col.names=TRUE, row.names=FALSE)

}
```


```{r}
# Get Z-score for each gene
idx.sub_D3_MP <- which(sample_plan_MP$Sheet=="MTX_PBS_D3")
diff_gene_normalized_counts_MP_D3 <- v$E[diff_genes_D3_MP$gene_id, idx.sub_D3_MP]
diff_gene_normalized_counts_MP_D3 <- diff_gene_normalized_counts_MP_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D3_MP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_MP_D3 <- diff_gene_normalized_counts_MP_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_MP_D3 <- top_20_up_down_diff_gene_normalized_counts_MP_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MP_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MP_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_MP_D3 <- sample_plan_MP %>% filter(sample_plan_MP$Day== "D3")
top_annot <- HeatmapAnnotation(df = sample_plan_MP_D3["Treatment"],
                               col = list(Treatment = c("MTX_D3" = "lightgreen", "PBS_D3" = "seagreen")))


# Heatmap - top 20 (up and down) regulated genes
png(file = heatmap_top_20_up_down_reg_genes, height = 800, width = 1200)
pmap_D3_MP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MP_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_MP_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_MP_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MP_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_MP_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D3_MP
dev.off()
```








## **3.1. MTX vs PBS** {.tabset .tabset-fade .tabset-pills}


### D3
```{r results_dea_MTX(D3)_vs_PBS(D3)}
get_results_dea(fit_contrast_MP, "MP_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D3.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_MTX_vs_PBS_D3.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_MTX_vs_PBS_D3.csv", "Volcano Plot: MTX(D3)_vs_PBS(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D3)_vs_PBS(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D3_MP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D3_MP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D3_MP.csv")
```

### D10
```{r results_dea_MTX(D10)_vs_PBS(D10)}
get_results_dea(fit_contrast_MP, "MP_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D10.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_MTX_vs_PBS_D10.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_MTX_vs_PBS_D10.csv", "Volcano Plot: MTX(D10)_vs_PBS(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_PBS(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_MP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_MP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_MP.csv")
```

### D22
```{r results_dea_MTX(D22)_vs_PBS(D22)}
get_results_dea(fit_contrast_MP, "MP_D22", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D22.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_MTX_vs_PBS_D22.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_MTX_vs_PBS_D22.csv", "/Volcano Plot: MTX(D22)_vs_PBS(D22)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_PBS(D22)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_MP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_MP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_MP.csv")
```




 
 



<!-- ```{r} -->
<!-- # Get Z-score for each gene -->
<!-- idx.sub_D3_MP <- which(sample_plan_MP$Sheet=="MTX_PBS_D3") -->
<!-- diff_gene_normalized_counts_MP_D3 <- v_MP$E[diff_genes_D3_MP$gene_id, idx.sub_D3_MP] -->
<!-- diff_gene_normalized_counts_MP_D3 <- diff_gene_normalized_counts_MP_D3 %>%  -->
<!--   t() %>%  -->
<!--   scale() %>% -->
<!--   t() %>% -->
<!--   bind_cols(diff_genes_D3_MP) -->


<!-- # Get top 20 up and down regulated genes -->
<!-- top_20_up_down_diff_gene_normalized_counts_MP_D3 <- diff_gene_normalized_counts_MP_D3 %>%  -->
<!--   dplyr::group_by(group) %>% -->
<!--   arrange(desc(logFC), adj.P.Val) %>% -->
<!--   slice(c(1:20)) %>% -->
<!--   as.data.frame() -->

<!-- # Get corresponding matrix for heatmap -->
<!-- matrix_top_20_up_down_diff_gene_normalized_counts_MP_D3 <- top_20_up_down_diff_gene_normalized_counts_MP_D3 %>%     -->
<!--   magrittr::set_rownames(make.unique(.$genes)) %>%  -->
<!--   dplyr::select(-(gene_id:group)) %>% -->
<!--   as.matrix() -->


<!-- # Create annotations -->
<!-- right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MP_D3$group , col = list(group = c("up" = "orange2", "down" = "gold"))) -->
<!-- right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MP_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2"))) -->
<!-- sample_plan_MP_D3 <- sample_plan_MP %>% filter(sample_plan_MP$Day== "D3") -->
<!-- top_annot <- HeatmapAnnotation(df = sample_plan_MP_D3["Treatment"], -->
<!--                                col = list(Treatment = c("MTX_D3" = "lightgreen", "PBS_D3" = "seagreen"))) -->


<!-- # Heatmap - top 20 (up and down) -->
<!-- png(file = "Heatmap: MTX(D3)_vs_PBS(D3).png", height = 800, width = 1200) -->
<!-- pmap_D3_MP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MP_D3, -->
<!--                                       name = "z-score",  -->
<!--                                       cluster_columns = TRUE, -->
<!--                                       cluster_column_slices = FALSE, -->
<!--                                       show_column_dend = FALSE, -->
<!--                                       column_split = sample_plan_MP_D3$Treatment, -->
<!--                                       row_split = top_20_up_down_diff_gene_normalized_counts_MP_D3$group, -->
<!--                                       row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MP_D3$group))), -->
<!--                                       cluster_rows = FALSE, -->
<!--                                       column_title = c(factor(levels(sample_plan_MP_D3$Treatment))), -->
<!--                                       clustering_method_columns = "ward.D2",  -->
<!--                                       right_annotation = c(right_annot1, right_annot2),  -->
<!--                                       column_names_gp = grid::gpar(fontsize = 10), -->
<!--                                       top_annotation = top_annot, -->
<!--                                       show_column_names = FALSE) -->


<!-- pmap_D3_MP -->
<!-- dev.off() -->

<!-- ``` -->




<!-- ```{r} -->

<!-- # Get Z-score for each gene -->
<!-- idx.sub_D10_MP <- which(sample_plan_MP$Sheet=="MTX_PBS_D10") -->
<!-- diff_gene_normalized_counts_MP_D10 <- v_MP$E[diff_genes_D10_MP$gene_id, idx.sub_D10_MP] -->
<!-- diff_gene_normalized_counts_MP_D10 <- diff_gene_normalized_counts_MP_D10 %>%  -->
<!--   t() %>%  -->
<!--   scale() %>% -->
<!--   t() %>% -->
<!--   bind_cols(diff_genes_D10_MP) -->


<!-- # Get top 20 up and down regulated genes -->
<!-- top_20_up_down_diff_gene_normalized_counts_MP_D10 <- diff_gene_normalized_counts_MP_D10 %>%  -->
<!--   dplyr::group_by(group) %>% -->
<!--   arrange(desc(logFC), adj.P.Val) %>% -->
<!--   slice(c(1:20)) %>% -->
<!--   as.data.frame() -->

<!-- # Get corresponding matrix for heatmap -->
<!-- matrix_top_20_up_down_diff_gene_normalized_counts_MP_D10 <- top_20_up_down_diff_gene_normalized_counts_MP_D10 %>%     -->
<!--   magrittr::set_rownames(make.unique(.$genes)) %>%  -->
<!--   dplyr::select(-(gene_id:group)) %>% -->
<!--   as.matrix() -->


<!-- # Create annotations -->
<!-- right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MP_D10$group , col = list(group = c("up" = "orange2", "down" = "gold"))) -->
<!-- right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MP_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2"))) -->
<!-- sample_plan_MP_D10 <- sample_plan_MP %>% filter(sample_plan_MP$Day== "D10") -->
<!-- top_annot <- HeatmapAnnotation(df = sample_plan_MP_D10["Treatment"], -->
<!--                                col = list(Treatment = c("MTX_D10" = "lightgreen", "PBS_D10" = "seagreen"))) -->


<!-- # Heatmap - top 20 (up and down) -->
<!-- png(file = "Heatmap: MTX(D10)_vs_PBS(D10).png", height = 800, width = 1200) -->
<!-- pmap_D10_MP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MP_D10, -->
<!--                                        name = "z-score",  -->
<!--                                        cluster_columns = TRUE, -->
<!--                                        cluster_column_slices = FALSE, -->
<!--                                        show_column_dend = FALSE, -->
<!--                                        column_split = sample_plan_MP_D10$Treatment, -->
<!--                                        row_split = top_20_up_down_diff_gene_normalized_counts_MP_D10$group, -->
<!--                                        row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MP_D10$group))), -->
<!--                                        cluster_rows = FALSE, -->
<!--                                        column_title = c(factor(levels(sample_plan_MP_D10$Treatment))), -->
<!--                                        clustering_method_columns = "ward.D2",  -->
<!--                                        right_annotation = c(right_annot1, right_annot2),  -->
<!--                                        column_names_gp = grid::gpar(fontsize = 10), -->
<!--                                        top_annotation = top_annot, -->
<!--                                        show_column_names = FALSE) -->


<!-- pmap_D10_MP -->
<!-- dev.off() -->


<!-- # Get common down (D3) and up (D10) protein-coding genes  -->
<!-- down_protein_coding_MP <- diff_genes_D3_MP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "down") -->
<!-- up_protein_coding_MP <- diff_genes_D10_MP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "up") -->
<!-- down_up_protein_coding_MP <- inner_join(down_protein_coding_MP, up_protein_coding_MP, by="gene_id")  -->
<!-- gene_names_down_up_protein_coding_MP  <- down_up_protein_coding_MP %>% pull("genes.x") %>% write.csv("common_genes_down_up_protein_coding_MP.csv") -->
<!-- #intersect(strsplit(str_replace_all(go_analysis_MTX_vs_PBS_D10_not_redundant_df[1,"geneID"], "/", " "), " ")[[1]], gene_names_down_up_protein_coding_MP) -->

<!-- # Create annotations -->
<!-- sample_plan_MP_D3_D10 <- sample_plan_MP %>% filter(sample_plan_MP$Day== "D3" | sample_plan_MP$Day== "D10") -->
<!-- top_annot_MP_D3_D10 <- HeatmapAnnotation(df = sample_plan_MP_D3_D10["Treatment"], -->
<!--                                col = list(Treatment = c("PBS_D3" = "seagreen", "MTX_D3" = "lightgreen", "PBS_D10" = "red", "MTX_D10" = "blue"))) -->

<!-- # Get counts for these common genes -->
<!-- idx.sub_D3_D10_MP <- which(sample_plan_MP_D3_D10$Sheet=="MTX_PBS_D3" | sample_plan_MP_D3_D10$Sheet=="MTX_PBS_D10") -->
<!-- normalized_counts_MP_D3_D10 <- v_MP$E[gene_names_down_up_protein_coding_MP_id, idx.sub_D3_D10_MP] -->
<!-- normalized_counts_MP_D3_D10 <- normalized_counts_MP_D3_D10 %>% as.data.frame() %>% rownames_to_column("gene_id") %>%  -->
<!--   inner_join(diff_genes_D10_MP[,c("genes","gene_id")], by="gene_id") %>% magrittr::set_rownames(make.unique(.$genes)) %>% dplyr::select(-c("gene_id","genes")) %>% t() %>% scale() %>%  t() %>% as.matrix() -->



<!-- sample_plan_MP_D3_D10$Treatment <- factor(sample_plan_MP_D3_D10$Treatment, levels = c("PBS_D3","MTX_D3","PBS_D10","MTX_D10")) -->


<!-- # Heatmap - common down (D3) and up (D10) protein-coding genes -->
<!-- png(file = "Heatmap: MTX(D3)_vs_PBS(D3)_vs_MTX(D10)_vs_PBS(D10).png", height = 800, width = 1200) -->
<!-- pmap_D3_D10_MP <- ComplexHeatmap::Heatmap(normalized_counts_MP_D3_D10, -->
<!--                                       name = "Counts",  -->
<!--                                       cluster_columns = FALSE, -->
<!--                                       cluster_column_slices = FALSE, -->
<!--                                       show_column_dend = FALSE, -->
<!--                                       column_split = sample_plan_MP_D3_D10$Treatment, -->
<!--                                       #row_split = c(down_up_protein_coding_MP$group.x, down_up_protein_coding_MP$group.y), -->
<!--                                       #row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MP_D3$group))), -->
<!--                                       cluster_rows = TRUE, -->
<!--                                       column_title = c(factor(levels(sample_plan_MP_D3_D10$Treatment))), -->
<!--                                       clustering_method_columns = "ward.D2",  -->
<!--                                       column_names_gp = grid::gpar(fontsize = 10, fontface="bold"), -->
<!--                                       row_names_gp = grid::gpar(fontsize = 10, fontface = "bold"), -->
<!--                                       top_annotation = top_annot_MP_D3_D10, -->
<!--                                       show_column_names = FALSE) -->

<!-- dev.off() -->
<!-- pmap_D3_D10_MP -->


<!-- ``` -->






## **3.2. OXA vs PBS** {.tabset .tabset-fade .tabset-pills}


### D3
```{r results_dea_Oxa(D3)_vs_PBS(D3)}
get_results_dea(fit_contrast_OP, "OP_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D3.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_OXA_vs_PBS_D3.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_OXA_vs_PBS_D3.csv","Volcano Plot: OXA(D3)_vs_PBS(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D3)_vs_PBS(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D3_OP.csv", "prop_by_type_diff_genes_D3_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D3_OP.csv")
```

### D10
```{r results_dea_Oxa(D10)_vs_PBS(D10)}
get_results_dea(fit_contrast_OP, "OP_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D10.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_OXA_vs_PBS_D10.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_OXA_vs_PBS_D10.csv", "Volcano Plot: OXA(D10)_vs_PBS(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D10)_vs_PBS(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_OP.csv", "prop_by_type_diff_genes_D10_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_OP.csv")
```

### D22
```{r results_dea_Oxa(D22)_vs_PBS(D22)}
get_results_dea(fit_contrast_OP, "OP_D22", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D22.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/up_gene_list_OXA_vs_PBS_D22.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/down_gene_list_OXA_vs_PBS_D22.csv", "Volcano Plot: OXA(D22)_vs_PBS(D22)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_OXA(D22)_vs_PBS(D22)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_OP.csv", "prop_by_type_diff_genes_D22_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_OP.csv")
```






<!-- ```{r} -->
<!-- # Get Z-score for each gene -->
<!-- idx.sub_D3_OP <- which(sample_plan_OCP$Treatment=="Oxa_D3" | sample_plan_OCP$Treatment=="PBS_D3" ) -->
<!-- diff_gene_normalized_counts_OP_D3 <- v_OCP$E[diff_genes_D3_OP$gene_id, idx.sub_D3_OP] -->
<!-- diff_gene_normalized_counts_OP_D3 <- diff_gene_normalized_counts_OP_D3 %>%  -->
<!--   t() %>%  -->
<!--   scale() %>% -->
<!--   t() %>% -->
<!--   bind_cols(diff_genes_D3_OP) -->


<!-- # Get top 20 up and down regulated genes -->
<!-- top_20_up_down_diff_gene_normalized_counts_OP_D3 <- diff_gene_normalized_counts_OP_D3 %>%  -->
<!--                                                     dplyr::group_by(group) %>% -->
<!--                                                     arrange(desc(logFC), adj.P.Val) %>% -->
<!--                                                     slice(c(1:20)) %>% -->
<!--                                                     as.data.frame() -->

<!-- # Get corresponding matrix for heatmap -->
<!-- matrix_top_20_up_down_diff_gene_normalized_counts_OP_D3 <- top_20_up_down_diff_gene_normalized_counts_OP_D3 %>%     -->
<!--                                                            magrittr::set_rownames(make.unique(.$genes)) %>%  -->
<!--                                                            dplyr::select(-(gene_id:group)) %>% -->
<!--                                                            as.matrix() -->


<!-- # Create annotations -->
<!-- right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_OP_D3$group , col = list(group = c("up" = "orange2", "down" = "gold"))) -->
<!-- right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_OP_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2"))) -->
<!-- sample_plan_OCP_D3 <- sample_plan_OCP %>% filter(Treatment== "Oxa_D3" | Treatment== "PBS_D3") -->
<!-- top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D3["Treatment"], -->
<!--                                col = list(Treatment = c("Oxa_D3" = "lightgreen", "PBS_D3" = "seagreen"))) -->


<!-- # Heatmap  -->
<!-- png(file = "Heatmap: Oxa(D3)_vs_PBS(D3).png", height = 800, width = 1200) -->
<!-- pmap_D3_OP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_OP_D3, -->
<!--                                       name = "z-score",  -->
<!--                                       cluster_columns = TRUE, -->
<!--                                       cluster_column_slices = FALSE, -->
<!--                                       show_column_dend = FALSE, -->
<!--                                       column_split = sample_plan_OCP_D3$Treatment, -->
<!--                                       row_split = top_20_up_down_diff_gene_normalized_counts_OP_D3$group, -->
<!--                                       row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_OP_D3$group))), -->
<!--                                       cluster_rows = FALSE, -->
<!--                                       column_title = c(factor(levels(sample_plan_OCP_D3$Treatment))), -->
<!--                                       clustering_method_columns = "ward.D2",  -->
<!--                                       right_annotation = c(right_annot1, right_annot2),  -->
<!--                                       column_names_gp = grid::gpar(fontsize = 10), -->
<!--                                       top_annotation = top_annot, -->
<!--                                       show_column_names = FALSE) -->


<!-- pmap_D3_OP -->
<!-- dev.off() -->


<!-- ``` -->


<!-- ```{r} -->

<!-- # Get Z-score for each gene -->
<!-- idx.sub_D10_OP <- which(sample_plan_OCP$Treatment == "Oxa_D10" | sample_plan_OCP$Treatment == "PBS_D10") -->
<!-- diff_gene_normalized_counts_OP_D10 <- v_OCP$E[diff_genes_D10_OP$gene_id, idx.sub_D10_OP] -->
<!-- diff_gene_normalized_counts_OP_D10 <- diff_gene_normalized_counts_OP_D10 %>%  -->
<!--   t() %>%  -->
<!--   scale() %>% -->
<!--   t() %>% -->
<!--   bind_cols(diff_genes_D10_OP) -->


<!-- # Get top 20 up and down regulated genes -->
<!-- top_20_up_down_diff_gene_normalized_counts_OP_D10 <- diff_gene_normalized_counts_OP_D10 %>%  -->
<!--                                                     dplyr::group_by(group) %>% -->
<!--                                                     arrange(desc(logFC), adj.P.Val) %>% -->
<!--                                                     slice(c(1:20)) %>% -->
<!--                                                     as.data.frame() -->

<!-- # Get corresponding matrix for heatmap -->
<!-- matrix_top_20_up_down_diff_gene_normalized_counts_OP_D10 <- top_20_up_down_diff_gene_normalized_counts_OP_D10 %>%     -->
<!--                                                            magrittr::set_rownames(make.unique(.$genes)) %>%  -->
<!--                                                            dplyr::select(-(gene_id:group)) %>% -->
<!--                                                            as.matrix() -->


<!-- # Create annotations -->
<!-- right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_OP_D10$group , col = list(group = c("up" = "orange2", "down" = "gold"))) -->
<!-- right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_OP_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2"))) -->
<!-- sample_plan_OCP_D10 <- sample_plan_OCP %>% filter(Treatment == "Oxa_D10" | Treatment == "PBS_D10") -->
<!-- top_annot_OP <- HeatmapAnnotation(df = sample_plan_OCP_D10["Treatment"], -->
<!--                                col = list(Treatment = c("Oxa_D10" = "lightgreen", "PBS_D10" = "seagreen"))) -->


<!-- # Heatmap  -->
<!-- png(file = "Heatmap: Oxa(D10)_vs_PBS(D10).png", height = 800, width = 1200) -->
<!-- pmap_D10_OP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_OP_D10, -->
<!--                                       name = "z-score",  -->
<!--                                       cluster_columns = TRUE, -->
<!--                                       cluster_column_slices = FALSE, -->
<!--                                       show_column_dend = FALSE, -->
<!--                                       column_split = sample_plan_OCP_D10$Treatment, -->
<!--                                       row_split = top_20_up_down_diff_gene_normalized_counts_OP_D10$group, -->
<!--                                       row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_OP_D10$group))), -->
<!--                                       cluster_rows = FALSE, -->
<!--                                       column_title = c(factor(levels(sample_plan_OCP_D10$Treatment))), -->
<!--                                       clustering_method_columns = "ward.D2",  -->
<!--                                       right_annotation = c(right_annot1, right_annot2),  -->
<!--                                       column_names_gp = grid::gpar(fontsize = 10), -->
<!--                                       top_annotation = top_annot, -->
<!--                                       show_column_names = FALSE) -->


<!-- pmap_D10_OP -->
<!-- dev.off() -->

<!-- # Get common down (D3) and up(D10) protein-coding genes  -->
<!-- down_protein_coding_OP <- diff_genes_D3_OP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "down") -->
<!-- up_protein_coding_OP <- diff_genes_D10_OP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "up") -->
<!-- down_up_protein_coding_OP <- inner_join(down_protein_coding_OP, up_protein_coding_OP, by="gene_id")  -->
<!-- gene_names_down_up_protein_coding_OP  <- down_up_protein_coding_OP %>% pull("genes.x") %>% write.csv("common_genes_down_up_protein_coding_OP.csv") -->


<!-- # Create annotations -->
<!-- sample_plan_OP_D3_D10 <- sample_plan_OCP %>% filter(sample_plan_OCP$Treatment== "Oxa_D3" | sample_plan_OCP$Treatment== "Oxa_D10" | sample_plan_OCP$Treatment== "PBS_D3" | sample_plan_OCP$Treatment== "PBS_D10") -->
<!-- top_annot_OP_D3_D10 <- HeatmapAnnotation(df = sample_plan_OP_D3_D10["Treatment"], -->
<!--                                col = list(Treatment = c("Oxa_D3" = "lightgreen", "PBS_D3" = "seagreen", "Oxa_D10" = "blue", "PBS_D10" = "red"))) -->

<!-- # Get counts for these common genes -->
<!-- idx.sub_D3_D10_OP <- which(sample_plan_OCP$Treatment== "Oxa_D3" | sample_plan_OCP$Treatment== "Oxa_D10" | sample_plan_OCP$Treatment== "PBS_D3" | sample_plan_OCP$Treatment== "PBS_D10") -->
<!-- gene_names_down_up_protein_coding_OP_id  <- down_up_protein_coding_OP %>% pull("gene_id") -->
<!-- normalized_counts_OP_D3_D10 <- v_OCP$E[gene_names_down_up_protein_coding_OP_id[1:50], idx.sub_D3_D10_OP] -->
<!-- normalized_counts_OP_D3_D10 <- normalized_counts_OP_D3_D10 %>% as.data.frame() %>% rownames_to_column("gene_id") %>%  -->
<!--   inner_join(diff_genes_D10_OP[,c("genes","gene_id")], by="gene_id") %>% magrittr::set_rownames(make.unique(.$genes)) %>% dplyr::select(-c("gene_id","genes")) %>% t() %>% scale() %>%  t() %>% as.matrix() -->



<!-- sample_plan_OP_D3_D10$Treatment <- factor(sample_plan_OP_D3_D10$Treatment, levels = c("PBS_D3","Oxa_D3", "PBS_D10", "Oxa_D10")) -->


<!-- # Heatmap - common down (D3) and up(D10) protein-coding genes -->
<!-- png(file = "Heatmap: Oxa(D3)_vs_PBS(D3)_vs_Oxa(D10)_vs_PBS(D10).png", height = 800, width = 1200) -->
<!-- pmap_D3_D10_OP <- ComplexHeatmap::Heatmap(normalized_counts_OP_D3_D10, -->
<!--                                       name = "Counts",  -->
<!--                                       cluster_columns = FALSE, -->
<!--                                       cluster_column_slices = FALSE, -->
<!--                                       show_column_dend = FALSE, -->
<!--                                       column_split = sample_plan_OP_D3_D10$Treatment, -->
<!--                                       cluster_rows = TRUE, -->
<!--                                       column_title = c(factor(levels(sample_plan_OP_D3_D10$Treatment))), -->
<!--                                       clustering_method_columns = "ward.D2",  -->
<!--                                       column_names_gp = grid::gpar(fontsize = 10, fontface="bold"), -->
<!--                                       row_names_gp = grid::gpar(fontsize = 10, fontface = "bold"), -->
<!--                                       top_annotation = top_annot_OP_D3_D10, -->
<!--                                       show_column_names = FALSE) -->

<!-- dev.off() -->
<!-- pmap_D3_D10_OP -->

<!-- n = list(intersect_OP = gene_names_down_up_protein_coding_OP, intersect_OP = gene_names_down_up_protein_coding_OP) -->
<!-- png(file = "Venn_list_overlap.png") -->
<!-- ggvenn(n, stroke_size = 0.5) -->
<!-- dev.off() -->
<!-- list_overlap <- intersect(gene_names_down_up_protein_coding_OP, gene_names_down_up_protein_coding_OP) -->
<!-- write.csv(list_overlap, "list_overlap.csv") -->
<!-- ``` -->



```{r}

# Get Z-score for each gene
idx.sub_D22_OP <- which(sample_plan_OCP$Treatment == "Oxa_D22" | sample_plan_OCP$Treatment == "PBS_D22")
diff_gene_normalized_counts_OP_D22 <- v_OCP$E[diff_genes_D22_OP$gene_id, idx.sub_D22_OP]
diff_gene_normalized_counts_OP_D22 <- diff_gene_normalized_counts_OP_D22 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_OP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_OP_D22 <- diff_gene_normalized_counts_OP_D22 %>% 
                                                    dplyr::group_by(group) %>%
                                                    arrange(desc(logFC), adj.P.Val) %>%
                                                    slice(c(1:20)) %>%
                                                    as.data.frame()
                                                    
# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_OP_D22 <- top_20_up_down_diff_gene_normalized_counts_OP_D22 %>%    
                                                           magrittr::set_rownames(make.unique(.$genes)) %>% 
                                                           dplyr::select(-(gene_id:group)) %>%
                                                           as.matrix()
                                 
  
# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_OP_D22$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_OP_D22$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22 <- sample_plan_OCP %>% filter(Treatment == "Oxa_D22" | Treatment == "PBS_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22["Treatment"],
                               col = list(Treatment = c("Oxa_D22" = "lightgreen", "PBS_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Oxa(D22)_vs_PBS(D22).png", height = 800, width = 1200)
pmap_D22_OP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_OP_D22,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_OP_D22$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_OP_D22$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_OP
dev.off()


```


## **3.3. Cis vs PBS** {.tabset .tabset-fade .tabset-pills}


### D3
```{r results_dea_Cis(D3)_vs_PBS(D3)}
get_results_dea(fit_contrast_CP, "CP_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D3.csv", "Volcano Plot: CIS(D3)_vs_PBS(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D3)_vs_PBS(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D3_OP.csv", "prop_by_type_diff_genes_D3_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D3_OP.csv")
```

### D10
```{r results_dea_Cis(D10)_vs_PBS(D10)}
get_results_dea(fit_contrast_CP, "CP_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D10.csv", "Volcano Plot: CIS(D10)_vs_PBS(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D10)_vs_PBS(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_OP.csv", "prop_by_type_diff_genes_D10_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_OP.csv")
```

### D22
```{r results_dea_Cis(D22)_vs_PBS(D22)}
get_results_dea(fit_contrast_CP, "CP_D22", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D22.csv", "Volcano Plot: CIS(D22)_vs_PBS(D22)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_CIS(D22)_vs_PBS(D22)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_OP.csv", "prop_by_type_diff_genes_D22_OP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_OP.csv")
```



```{r}

# Get Z-score for each gene
idx.sub_D3_CP <- which(sample_plan_OCP$Treatment =="Cis_D3" | sample_plan_OCP$Treatment =="PBS_D3")
diff_gene_normalized_counts_CP_D3 <- v_OCP$E[diff_genes_D3_CP$gene_id, idx.sub_D3_CP]
diff_gene_normalized_counts_CP_D3 <- diff_gene_normalized_counts_CP_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D3_CP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_CP_D3 <- diff_gene_normalized_counts_CP_D3 %>% 
                                                    dplyr::group_by(group) %>%
                                                    arrange(desc(logFC), adj.P.Val) %>%
                                                    slice(c(1:20)) %>%
                                                    as.data.frame()
                                                    
# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_CP_D3 <- top_20_up_down_diff_gene_normalized_counts_CP_D3 %>%    
                                                           magrittr::set_rownames(make.unique(.$genes)) %>% 
                                                           dplyr::select(-(gene_id:group)) %>%
                                                           as.matrix()
                                 
  
# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_CP_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_CP_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D3 <- sample_plan_OCP %>% filter(Treatment == "Cis_D3" | Treatment == "PBS_D3")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D3["Treatment"],
                               col = list(Treatment = c("Cis_D3" = "lightgreen", "PBS_D3" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Cis(D3)_vs_PBS(D3).png", height = 800, width = 1200)
pmap_D3_CP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_CP_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_CP_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_CP_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D3_CP
dev.off()


```



```{r}

# Get Z-score for each gene
idx.sub_D10_CP <- which(sample_plan_OCP$Treatment =="Cis_D10" | sample_plan_OCP$Treatment =="PBS_D10")
diff_gene_normalized_counts_CP_D10 <- v_OCP$E[diff_genes_D10_CP$gene_id, idx.sub_D10_CP]
diff_gene_normalized_counts_CP_D10 <- diff_gene_normalized_counts_CP_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_CP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_CP_D10 <- diff_gene_normalized_counts_CP_D10 %>% 
                                                    dplyr::group_by(group) %>%
                                                    arrange(desc(logFC), adj.P.Val) %>%
                                                    slice(c(1:20)) %>%
                                                    as.data.frame()
                                                    
# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_CP_D10 <- top_20_up_down_diff_gene_normalized_counts_CP_D10 %>%    
                                                           magrittr::set_rownames(make.unique(.$genes)) %>% 
                                                           dplyr::select(-(gene_id:group)) %>%
                                                           as.matrix()
                                 
  
# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_CP_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_CP_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D10 <- sample_plan_OCP %>% filter(Treatment == "Cis_D10" | Treatment == "PBS_D10" )
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D10["Treatment"],
                               col = list(Treatment = c("Cis_D10" = "lightgreen", "PBS_D10" = "seagreen")))


# Heatmap
png(file = "Heatmap: Cis(D10)_vs_PBS(D10).png", height = 800, width = 1200)
pmap_D10_CP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_CP_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_CP_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_CP_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_CP
dev.off()


# Get common down (D3) and up(D10) protein-coding genes 
down_protein_coding_CP <- diff_genes_D3_CP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "down")
up_protein_coding_CP <- diff_genes_D10_CP %>%  filter(gene_type_summary == "protein_coding_gene" & group == "up")
down_up_protein_coding_CP <- inner_join(down_protein_coding_CP, up_protein_coding_CP, by="gene_id") 
gene_names_down_up_protein_coding_CP  <- down_up_protein_coding_CP %>% pull("genes.x") %>% write.csv("common_genes_down_up_protein_coding_CP.csv")


# Create annotations
sample_plan_CP_D3_D10 <- sample_plan_OCP %>% filter(sample_plan_OCP$Treatment== "Cis_D3" | sample_plan_OCP$Treatment== "Cis_D10" | sample_plan_OCP$Treatment== "PBS_D3" | sample_plan_OCP$Treatment== "PBS_D10")
top_annot_CP_D3_D10 <- HeatmapAnnotation(df = sample_plan_CP_D3_D10["Treatment"],
                               col = list(Treatment = c("Cis_D3" = "lightgreen", "PBS_D3" = "seagreen", "Cis_D10" = "blue", "PBS_D10" = "red")))

# Get counts for these common genes
idx.sub_D3_D10_CP <- which(sample_plan_OCP$Treatment== "Cis_D3" | sample_plan_OCP$Treatment== "Cis_D10" | sample_plan_OCP$Treatment== "PBS_D3" | sample_plan_OCP$Treatment== "PBS_D10")
gene_names_down_up_protein_coding_CP_id  <- down_up_protein_coding_CP %>% pull("gene_id")
normalized_counts_CP_D3_D10 <- v_OCP$E[gene_names_down_up_protein_coding_CP_id, idx.sub_D3_D10_CP]
normalized_counts_CP_D3_D10 <- normalized_counts_CP_D3_D10 %>% as.data.frame() %>% rownames_to_column("gene_id") %>% 
  inner_join(diff_genes_D10_CP[,c("genes","gene_id")], by="gene_id") %>% magrittr::set_rownames(make.unique(.$genes)) %>% dplyr::select(-c("gene_id","genes")) %>% t() %>% scale() %>%  t() %>% as.matrix()



sample_plan_CP_D3_D10$Treatment <- factor(sample_plan_CP_D3_D10$Treatment, levels = c("PBS_D3","Cis_D3", "PBS_D10", "Cis_D10"))


# Heatmap - common down (D3) and up(D10) protein-coding genes
png(file = "Heatmap: Cis(D3)_vs_PBS(D3)_vs_Cis(D10)_vs_PBS(D10).png", height = 800, width = 1200)
pmap_D3_D10_CP <- ComplexHeatmap::Heatmap(normalized_counts_CP_D3_D10,
                                      name = "Counts", 
                                      cluster_columns = FALSE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_CP_D3_D10$Treatment,
                                      cluster_rows = TRUE,
                                      column_title = c(factor(levels(sample_plan_CP_D3_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      column_names_gp = grid::gpar(fontsize = 10, fontface="bold"),
                                      row_names_gp = grid::gpar(fontsize = 10, fontface = "bold"),
                                      top_annotation = top_annot_CP_D3_D10,
                                      show_column_names = FALSE)

dev.off()
pmap_D3_D10_CP



```



```{r}


# Get Z-score for each gene
idx.sub_D10_D3_MM <- which(sample_plan_OP$Treatment == "MTX_D3" | sample_plan_OP$Treatment == "MTX_D10")
diff_gene_normalized_counts_MTX_D10_D3 <- v_OP$E[diff_genes_D10_D3_MM$gene_id, idx.sub_D10_D3_MM]
diff_gene_normalized_counts_MTX_D10_D3 <- diff_gene_normalized_counts_MTX_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_MM)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 <- diff_gene_normalized_counts_MTX_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D10_D3 <- sample_plan_OP %>% filter(Treatment == "MTX_D3" | Treatment == "MTX_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D10_D3["Treatment"],
                               col = list(Treatment = c("MTX_D3" = "lightgreen", "MTX_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: MTX(D10)_vs_MTX(D3).png", height = 800, width = 1200)
pmap_D10_D3_MM <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_MM
dev.off()

```

## **3.3. MTX vs MTX** {.tabset .tabset-fade .tabset-pills}


### D22 vs D10
```{r results_dea_MTX(D22)_vs_MTX(D10)}
get_results_dea(fit_contrast_MM, "MTX_D22_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D22_D10.csv", "Volcano Plot: MTX(D22)_vs_MTX(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_MM.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_MM_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D10_MM.csv")
```

### D22 vs D3
```{r results_dea_MTX(D22)_vs_MTX(D3)}
get_results_dea(fit_contrast_MM, "MTX_D22_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D22_D3.csv", "Volcano Plot: MTX(D22)_vs_MTX(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D22)_vs_MTX(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_MM.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_MM_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D3_MM.csv")
```

### D10 vs D3
```{r results_dea_MTX(D10)_vs_MTX(D3)}
get_results_dea(fit_contrast_MM, "MTX_D10_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D10_D3.csv", "Volcano Plot: MTX(D10)_vs_MTX(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_MTX(D10)_vs_MTX(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_MM.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_MM_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_D3_MM.csv")
```




```{r}
# Get Z-score for each gene
idx.sub_D22_D10_MM <- which(sample_plan_OP$Treatment == "MTX_D22" | sample_plan_OP$Treatment == "MTX_D10")
diff_gene_normalized_counts_MTX_D22_D10 <- v_OP$E[diff_genes_D22_D10_MM$gene_id, idx.sub_D22_D10_MM]
diff_gene_normalized_counts_MTX_D22_D10 <- diff_gene_normalized_counts_MTX_D22_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D10_MM)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10 <- diff_gene_normalized_counts_MTX_D22_D10 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10 <- top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D22_D10 <- sample_plan_OP %>% filter(Treatment == "MTX_D10" | Treatment == "MTX_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D22_D10["Treatment"],
                               col = list(Treatment = c("MTX_D10" = "lightgreen", "MTX_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: MTX(D22)_vs_MTX(D10).png", height = 800, width = 1200)
pmap_D22_D10_MM <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D22_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MTX_D22_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D22_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D10_MM
dev.off()

```


```{r}
# Get Z-score for each gene
idx.sub_D22_D3_MM <- which(sample_plan_OP$Treatment == "MTX_D22" | sample_plan_OP$Treatment == "MTX_D3")
diff_gene_normalized_counts_MTX_D22_D3 <- v_OP$E[diff_genes_D22_D3_MM$gene_id, idx.sub_D22_D3_MM]
diff_gene_normalized_counts_MTX_D22_D3 <- diff_gene_normalized_counts_MTX_D22_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D3_MM)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3 <- diff_gene_normalized_counts_MTX_D22_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3 <- top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D22_D3 <- sample_plan_OP %>% filter(Treatment == "MTX_D3" | Treatment == "MTX_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D22_D3["Treatment"],
                               col = list(Treatment = c("MTX_D3" = "lightgreen", "MTX_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: MTX(D22)_vs_MTX(D3).png", height = 800, width = 1200)
pmap_D22_D3_MM <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D22_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MTX_D22_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D22_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D3_MM
dev.off()

```


```{r}
# Get Z-score for each gene
idx.sub_D10_D3_MM <- which(sample_plan_OP$Treatment == "MTX_D10" | sample_plan_OP$Treatment == "MTX_D3")
diff_gene_normalized_counts_MTX_D10_D3 <- v_OP$E[diff_genes_D10_D3_MM$gene_id, idx.sub_D10_D3_MM]
diff_gene_normalized_counts_MTX_D10_D3 <- diff_gene_normalized_counts_MTX_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_MM)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 <- diff_gene_normalized_counts_MTX_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D10_D3 <- sample_plan_OP %>% filter(Treatment == "MTX_D3" | Treatment == "MTX_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D10_D3["Treatment"],
                               col = list(Treatment = c("MTX_D3" = "lightgreen", "MTX_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: MTX(D10)_vs_MTX(D3).png", height = 800, width = 1200)
pmap_D10_D3_MM <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_MTX_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_MM
dev.off()

```






## **3.4. Oxa vs Oxa** {.tabset .tabset-fade .tabset-pills}


### D22 vs D10
```{r results_dea_Oxa(D22)_vs_Oxa(D10)}
get_results_dea(fit_contrast_OO, "Oxa_D22_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D22_D10.csv", "Volcano Plot: Oxa(D22)_vs_Oxa(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_OO.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_OO_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D10_OO.csv")
```

### D22 vs D3
```{r results_dea_Oxa(D22)_vs_Oxa(D3)}
get_results_dea(fit_contrast_OO, "Oxa_D22_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D22_D3.csv", "Volcano Plot: Oxa(D22)_vs_Oxa(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D22)_vs_Oxa(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_OO.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_OO_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D3_OO.csv")
```

### D10 vs D3
```{r results_dea_Oxa(D10)_vs_Oxa(D3)}
get_results_dea(fit_contrast_OO, "Oxa_D10_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D10_D3.csv", "Volcano Plot: Oxa(D10)_vs_Oxa(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Oxa(D10)_vs_Oxa(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_OO.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_OO_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_D3_OO.csv")
```




```{r}
# Get Z-score for each gene
idx.sub_D10_D3_OO <- which(sample_plan_OCP$Treatment == "Oxa_D3" | sample_plan_OCP$Treatment == "Oxa_D10")
diff_gene_normalized_counts_Oxa_D10_D3 <- v_OCP$E[diff_genes_D10_D3_OO$gene_id, idx.sub_D10_D3_OO]
diff_gene_normalized_counts_Oxa_D10_D3 <- diff_gene_normalized_counts_Oxa_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_OO)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3 <- diff_gene_normalized_counts_Oxa_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D10_D3 <- sample_plan_OCP %>% filter(Treatment == "Oxa_D3" | Treatment == "Oxa_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D10_D3["Treatment"],
                               col = list(Treatment = c("Oxa_D3" = "lightgreen", "Oxa_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Oxa(D10)_vs_Oxa(D3).png", height = 800, width =1200)
pmap_D10_D3_OO <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Oxa_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_OO
dev.off()

```




```{r}
# Get Z-score for each gene
idx.sub_D22_D10_OO <- which(sample_plan_OCP$Treatment == "Oxa_D10" | sample_plan_OCP$Treatment == "Oxa_D22")
diff_gene_normalized_counts_Oxa_D22_D10 <- v_OCP$E[diff_genes_D22_D10_OO$gene_id, idx.sub_D22_D10_OO]
diff_gene_normalized_counts_Oxa_D22_D10 <- diff_gene_normalized_counts_Oxa_D22_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D10_OO)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10 <- diff_gene_normalized_counts_Oxa_D22_D10 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10 <- top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D10 <- sample_plan_OCP %>% filter(Treatment == "Oxa_D10" | Treatment == "Oxa_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D10["Treatment"],
                               col = list(Treatment = c("Oxa_D10" = "lightgreen", "Oxa_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Oxa(D22)_vs_Oxa(D10).png", height = 800, width = 1200)
pmap_D22_D10_OO <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D10_OO
dev.off()

```


```{r}
# Get Z-score for each gene
idx.sub_D22_D3_OO <- which(sample_plan_OCP$Treatment == "Oxa_D3" | sample_plan_OCP$Treatment == "Oxa_D22")
diff_gene_normalized_counts_Oxa_D22_D3 <- v_OCP$E[diff_genes_D22_D3_OO$gene_id, idx.sub_D22_D3_OO]
diff_gene_normalized_counts_Oxa_D22_D3 <- diff_gene_normalized_counts_Oxa_D22_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D3_OO)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3 <- diff_gene_normalized_counts_Oxa_D22_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3 <- top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D3 <- sample_plan_OCP %>% filter(Treatment == "Oxa_D3" | Treatment == "Oxa_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D3["Treatment"],
                               col = list(Treatment = c("Oxa_D3" = "lightgreen", "Oxa_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Oxa(D22)_vs_Oxa(D3).png", height = 800, width = 1200)
pmap_D22_D3_OO <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Oxa_D22_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D3_OO
dev.off()

```



## **3.5. Cis vs Cis** {.tabset .tabset-fade .tabset-pills}

### D22 vs D10
```{r results_dea_Cis(D22)_vs_Cis(D10)}
get_results_dea(fit_contrast_CC, "Cis_D22_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D22_D10.csv", "Volcano Plot: Cis(D22)_vs_Cis(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_CC.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_CC_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D10_CC.csv")
```

### D22 vs D3
```{r results_dea_Cis(D22)_vs_Cis(D3)}
get_results_dea(fit_contrast_CC, "Cis_D22_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D22_D3.csv", "Volcano Plot: Cis(D22)_vs_Cis(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D22)_vs_Cis(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_CC.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_CC_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D3_CC.csv")
```

### D10 vs D3
```{r results_dea_Cis(D10)_vs_Cis(D3)}
get_results_dea(fit_contrast_CC, "Cis_D10_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D10_D3.csv", "Volcano Plot: Cis(D10)_vs_Cis(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_Cis(D10)_vs_Cis(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_CC.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_CC_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_D3_CC.csv")
```



```{r}
# Get Z-score for each gene
idx.sub_D10_D3_CC <- which(sample_plan_OCP$Treatment == "Cis_D3" | sample_plan_OCP$Treatment == "Cis_D10")
diff_gene_normalized_counts_Cis_D10_D3 <- v_OCP$E[diff_genes_D10_D3_CC$gene_id, idx.sub_D10_D3_CC]
diff_gene_normalized_counts_Cis_D10_D3 <- diff_gene_normalized_counts_Cis_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_CC)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3 <- diff_gene_normalized_counts_Cis_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D10_D3 <- sample_plan_OCP %>% filter(Treatment == "Cis_D3" | Treatment == "Cis_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D10_D3["Treatment"],
                               col = list(Treatment = c("Cis_D3" = "lightgreen", "Cis_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Cis(D10)_vs_Cis(D3).png", height = 800, width = 1200)
pmap_D10_D3_CC <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Cis_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_CC
dev.off()

```



```{r}
# Get Z-score for each gene
idx.sub_D22_D10_CC <- which(sample_plan_OCP$Treatment == "Cis_D10" | sample_plan_OCP$Treatment == "Cis_D22")
diff_gene_normalized_counts_Cis_D22_D10 <- v_OCP$E[diff_genes_D22_D10_CC$gene_id, idx.sub_D22_D10_CC]
diff_gene_normalized_counts_Cis_D22_D10 <- diff_gene_normalized_counts_Cis_D22_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D10_CC)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10 <- diff_gene_normalized_counts_Cis_D22_D10 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10 <- top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D10 <- sample_plan_OCP %>% filter(Treatment == "Cis_D10" | Treatment == "Cis_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D10["Treatment"],
                               col = list(Treatment = c("Cis_D10" = "lightgreen", "Cis_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Cis(D22)_vs_Cis(D10).png", height = 800, width = 1200)
pmap_D22_D10_CC <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Cis_D22_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D10_CC
dev.off()

```




```{r}
# Get Z-score for each gene
idx.sub_D22_D3_CC <- which(sample_plan_OCP$Treatment == "Cis_D3" | sample_plan_OCP$Treatment == "Cis_D22")
diff_gene_normalized_counts_Cis_D22_D3 <- v_OCP$E[diff_genes_D22_D3_CC$gene_id, idx.sub_D22_D3_CC]
diff_gene_normalized_counts_Cis_D22_D3 <- diff_gene_normalized_counts_Cis_D22_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D3_CC)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3 <- diff_gene_normalized_counts_Cis_D22_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3 <- top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D3 <- sample_plan_OCP %>% filter(Treatment == "Cis_D3" | Treatment == "Cis_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D3["Treatment"],
                               col = list(Treatment = c("Cis_D3" = "lightgreen", "Cis_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: Cis(D22)_vs_Cis(D3).png", height = 800, width = 1200)
pmap_D22_D3_CC <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_Cis_D22_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D3_CC
dev.off()

```


## **3.6. PBS_MTX vs PBS_MTX** {.tabset .tabset-fade .tabset-pills}


### D22 vs D10
```{r results_dea_PBS_MTX(D22)_vs PBS_MTX(D10)}
get_results_dea(fit_contrast_PP_MTX, "PBS_D22_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_MTX_vs_PBS_MTX_D22_D10.csv", "Volcano Plot: PBS_MTX(D22)_vs_PBS_MTX(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D10)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_PP_MTX.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_PP_MTX_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D10_PP_MTX.csv")
```

### D22 vs D3
```{r results_dea_PBS_MTX(D22)_vs PBS_MTX(D3)}
get_results_dea(fit_contrast_PP_MTX, "PBS_D22_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_MTX_vs_PBS_MTX_D22_D3.csv", "Volcano Plot: PBS_MTX(D22)_vs_PBS_MTX(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D22)_vs_PBS_MTX(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_PP_MTX.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_PP_MTX_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D3_PP_MTX.csv")
```

### D10 vs D3
```{r results_dea_PBS_MTX(D10)_vs PBS_MTX(D3)}
get_results_dea(fit_contrast_PP_MTX, "PBS_D10_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_MTX_vs_PBS_MTX_D10_D3.csv", "Volcano Plot: PBS_MTX(D10)_vs_PBS_MTX(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_MTX(D10)_vs_PBS_MTX(D3)_colored_by_non_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_PP_MTX.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_PP_MTX_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_D3_PP_MTX.csv")
```




```{r}

# Get Z-score for each gene
idx.sub_D10_D3_PP_MTX <- which(sample_plan_OP$Treatment == "PBS_D3" | sample_plan_OP$Treatment == "PBS_D10")
diff_gene_normalized_counts_PBS_MTX_D10_D3 <- v_OP$E[diff_genes_D10_D3_PP_MTX$gene_id, idx.sub_D10_D3_PP_MTX]
diff_gene_normalized_counts_PBS_MTX_D10_D3 <- diff_gene_normalized_counts_PBS_MTX_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_PP_MTX)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3 <- diff_gene_normalized_counts_PBS_MTX_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D10_D3 <- sample_plan_OP %>% filter(Treatment == "PBS_D3" | Treatment == "PBS_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D10_D3["Treatment"],
                               col = list(Treatment = c("PBS_D3" = "lightgreen", "PBS_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_MTX(D10)_vs_PBS_MTX(D3).png", height = 800, width = 1200)
pmap_D10_D3_PP_MTX <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_PP_MTX
dev.off()

```



```{r}

# Get Z-score for each gene
idx.sub_D22_D10_PP_MTX <- which(sample_plan_OP$Treatment == "PBS_D10" | sample_plan_OP$Treatment == "PBS_D22")
diff_gene_normalized_counts_PBS_MTX_D22_D10 <- v_OP$E[diff_genes_D22_D10_PP_MTX$gene_id, idx.sub_D22_D10_PP_MTX]
diff_gene_normalized_counts_PBS_MTX_D22_D10 <- diff_gene_normalized_counts_PBS_MTX_D22_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D10_PP_MTX)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10 <- diff_gene_normalized_counts_PBS_MTX_D22_D10 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10 <- top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D22_D10 <- sample_plan_OP %>% filter(Treatment == "PBS_D10" | Treatment == "PBS_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D22_D10["Treatment"],
                               col = list(Treatment = c("PBS_D10" = "lightgreen", "PBS_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_MTX(D22)_vs_PBS_MTX(D10).png", height = 800, width = 1200)
pmap_D22_D10_PP_MTX <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D22_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D22_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D10_PP_MTX
dev.off()

```



```{r}
# Get Z-score for each gene
idx.sub_D22_D3_PP_MTX <- which(sample_plan_OP$Treatment == "PBS_D3" | sample_plan_OP$Treatment == "PBS_D22")
diff_gene_normalized_counts_PBS_MTX_D22_D3 <- v_OP$E[diff_genes_D22_D3_PP_MTX$gene_id, idx.sub_D22_D3_PP_MTX]
diff_gene_normalized_counts_PBS_MTX_D22_D3 <- diff_gene_normalized_counts_PBS_MTX_D22_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D3_PP_MTX)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3 <- diff_gene_normalized_counts_PBS_MTX_D22_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3 <- top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OP_D22_D3 <- sample_plan_OP %>% filter(Treatment == "PBS_D3" | Treatment == "PBS_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OP_D22_D3["Treatment"],
                               col = list(Treatment = c("PBS_D3" = "lightgreen", "PBS_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_MTX(D22)_vs_PBS_MTX(D3).png", height = 800, width = 1200)
pmap_D22_D3_PP_MTX <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OP_D22_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_MTX_D22_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OP_D22_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D3_PP_MTX
dev.off()

```


## **3.7. PBS_OCP vs PBS_OCP** {.tabset .tabset-fade .tabset-pills}


### D22 vs D10
```{r results_dea_PBS_OCP(D22)_vs PBS_OCP(D10)}
get_results_dea(fit_contrast_PP_OCP, "PBS_D22_D10", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_OCP_vs_PBS_OCP_D22_D10.csv", "Volcano Plot: PBS_OCP(D22)_vs_PBS_OCP(D10)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D10)_colored_by_non_coding.html", "prop_by_type_diff_genes_D22_D10_PP_OCP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D10_PP_OCP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D10_PP_OCP.csv")
```

### D22 vs D3
```{r results_dea_PBS_OCP(D22)_vs PBS_OCP(D3)}
get_results_dea(fit_contrast_PP_OCP, "PBS_D22_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_OCP_vs_PBS_OCP_D22_D3.csv", "Volcano Plot: PBS_OCP(D22)_vs_PBS_OCP(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D22)_vs_PBS_OCP(D3)_colored_by_non_coding.html", "prop_by_type_diff_genes_D22_D3_PP_OCP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D22_D3_PP_OCP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D22_D3_PP_OCP.csv")
```

### D10 vs D3
```{r results_dea_PBS_OCP(D10)_vs PBS_OCP(D3)}
get_results_dea(fit_contrast_PP_OCP, "PBS_D10_D3", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_OCP_vs_PBS_OCP_D10_D3.csv", "Volcano Plot: PBS_OCP(D10)_vs_PBS_OCP(D3)", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_logFC_and_adj.P.val.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_non_coding.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_logFC_and_adj.P.val.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_coding.html", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Volcano_plot_PBS_OCP(D10)_vs_PBS_OCP(D3)_colored_by_non_coding.html", "prop_by_type_diff_genes_D10_D3_PP_OCP.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/prop_by_type_diff_genes_D10_D3_PP_OCP_barplot.png", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/proportion_of_up_down_reg_by_type_D10_D3_PP_OCP.csv")
```




```{r}

# Get Z-score for each gene
idx.sub_D10_D3_PP_OCP <- which(sample_plan_OCP$Treatment == "PBS_D3" | sample_plan_OCP$Treatment == "PBS_D10")
diff_gene_normalized_counts_PBS_OCP_D10_D3 <- v_OCP$E[diff_genes_D10_D3_PP_OCP$gene_id, idx.sub_D10_D3_PP_OCP]
diff_gene_normalized_counts_PBS_OCP_D10_D3 <- diff_gene_normalized_counts_PBS_OCP_D10_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D10_D3_PP_OCP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3 <- diff_gene_normalized_counts_PBS_OCP_D10_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3 <- top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D10_D3 <- sample_plan_OCP %>% filter(Treatment == "PBS_D3" | Treatment == "PBS_D10")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D10_D3["Treatment"],
                               col = list(Treatment = c("PBS_D3" = "lightgreen", "PBS_D10" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_OCP(D10)_vs_PBS_OCP(D3).png", height = 800, width = 1200)
pmap_D10_D3_PP_OCP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D10_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D10_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D10_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D10_D3_PP_OCP
dev.off()

```



```{r}
# Get Z-score for each gene
idx.sub_D22_D10_PP_OCP <- which(sample_plan_OCP$Treatment == "PBS_D10" | sample_plan_OCP$Treatment == "PBS_D22")
diff_gene_normalized_counts_PBS_OCP_D22_D10 <- v_OCP$E[diff_genes_D22_D10_PP_OCP$gene_id, idx.sub_D22_D10_PP_OCP]
diff_gene_normalized_counts_PBS_OCP_D22_D10 <- diff_gene_normalized_counts_PBS_OCP_D22_D10 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D10_PP_OCP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10 <- diff_gene_normalized_counts_PBS_OCP_D22_D10 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10 <- top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D10 <- sample_plan_OCP %>% filter(Treatment == "PBS_D10" | Treatment == "PBS_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D10["Treatment"],
                               col = list(Treatment = c("PBS_D10" = "lightgreen", "PBS_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_OCP(D22)_vs_PBS_OCP(D10).png", height = 800, width = 1200)
pmap_D22_D10_PP_OCP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D10$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D10$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D10$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D10_PP_OCP
dev.off()

```



```{r}
# Get Z-score for each gene
idx.sub_D22_D3_PP_OCP <- which(sample_plan_OCP$Treatment == "PBS_D3" | sample_plan_OCP$Treatment == "PBS_D22")
diff_gene_normalized_counts_PBS_OCP_D22_D3 <- v_OCP$E[diff_genes_D22_D3_PP_OCP$gene_id, idx.sub_D22_D3_PP_OCP]
diff_gene_normalized_counts_PBS_OCP_D22_D3 <- diff_gene_normalized_counts_PBS_OCP_D22_D3 %>% 
  t() %>% 
  scale() %>%
  t() %>%
  bind_cols(diff_genes_D22_D3_PP_OCP)


# Get top 20 up and down regulated genes
top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3 <- diff_gene_normalized_counts_PBS_OCP_D22_D3 %>% 
  dplyr::group_by(group) %>%
  arrange(desc(logFC), adj.P.Val) %>%
  slice(c(1:20)) %>%
  as.data.frame()

# Get corresponding matrix for heatmap
matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3 <- top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3 %>%    
  magrittr::set_rownames(make.unique(.$genes)) %>% 
  dplyr::select(-(gene_id:group)) %>%
  as.matrix()


# Create annotations
right_annot1 = rowAnnotation(group = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3$group , col = list(group = c("up" = "orange2", "down" = "gold")))
right_annot2 = rowAnnotation(gene_type = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3$gene_type_summary , col = list(gene_type = c("non_coding_gene (not mitochondrial)" = "purple", "non_coding_gene (mitochondrial)" = "grey", "protein_coding_gene" = "green2")))
sample_plan_OCP_D22_D3 <- sample_plan_OCP %>% filter(Treatment == "PBS_D3" | Treatment == "PBS_D22")
top_annot <- HeatmapAnnotation(df = sample_plan_OCP_D22_D3["Treatment"],
                               col = list(Treatment = c("PBS_D3" = "lightgreen", "PBS_D22" = "seagreen")))


# Heatmap 
png(file = "Heatmap: PBS_OCP(D22)_vs_PBS_OCP(D3).png", height = 800, width = 1200)
pmap_D22_D3_PP_OCP <- ComplexHeatmap::Heatmap(matrix_top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3,
                                      name = "z-score", 
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = sample_plan_OCP_D22_D3$Treatment,
                                      row_split = top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3$group,
                                      row_title = c(factor(levels(top_20_up_down_diff_gene_normalized_counts_PBS_OCP_D22_D3$group))),
                                      cluster_rows = FALSE,
                                      column_title = c(factor(levels(sample_plan_OCP_D22_D3$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      right_annotation = c(right_annot1, right_annot2), 
                                      column_names_gp = grid::gpar(fontsize = 10),
                                      top_annotation = top_annot,
                                      show_column_names = FALSE)


pmap_D22_D3_PP_OCP
dev.off()

```


# **3.8. Overlapping differentially expressed genes** 



```{r}
# Load diff_genes : MTX vs PBS
gene_list_MTX_vs_PBS_D3 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D3.csv") %>% dplyr::select(-"X") 
gene_list_MTX_vs_PBS_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D10.csv") %>% dplyr::select(-"X") 
gene_list_MTX_vs_PBS_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D22.csv")%>% dplyr::select(-"X") 

# Load diff_genes : Oxa vs PBS
gene_list_Oxa_vs_PBS_D3 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D3.csv") %>% dplyr::select(-"X") 
gene_list_Oxa_vs_PBS_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D10.csv") %>% dplyr::select(-"X") 
gene_list_Oxa_vs_PBS_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D22.csv") %>% dplyr::select(-"X") 

# Load diff_genes : Cis vs PBS
gene_list_Cis_vs_PBS_D3 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D3.csv") %>% dplyr::select(-"X") 
gene_list_Cis_vs_PBS_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D10.csv") %>% dplyr::select(-"X")
gene_list_Cis_vs_PBS_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D22.csv") %>% dplyr::select(-"X")

# Load diff_genes : MTX vs MTX
gene_list_MTX_vs_MTX_D3_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D3_D10.csv") %>% dplyr::select(-"X") 
gene_list_MTX_vs_MTX_D10_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D10_D22.csv") %>% dplyr::select(-"X")
gene_list_MTX_vs_MTX_D3_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_MTX_D3_D22.csv") %>% dplyr::select(-"X")

# Load diff_genes : Oxa vs Oxa
gene_list_Oxa_vs_Oxa_D3_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D3_D10.csv") %>% dplyr::select(-"X") 
gene_list_Oxa_vs_Oxa_D10_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D10_D22.csv") %>% dplyr::select(-"X")
gene_list_Oxa_vs_Oxa_D3_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Oxa_vs_Oxa_D3_D22.csv") %>% dplyr::select(-"X")

# Load diff_genes : Cis vs Cis
gene_list_Cis_vs_Cis_D3_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D3_D10.csv") %>% dplyr::select(-"X") 
gene_list_Cis_vs_Cis_D10_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D10_D22.csv") %>% dplyr::select(-"X")
gene_list_Cis_vs_Cis_D3_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_Cis_vs_Cis_D3_D22.csv") %>% dplyr::select(-"X")

# Load diff_genes : PBS VS PBS (MTX)
gene_list_PBS_vs_PBS_MTX_D10_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_vs_PBS_D10_D22.csv") %>% dplyr::select(-"X")
gene_list_PBS_vs_PBS_MTX_D3_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_vs_PBS_D3_D10.csv") %>% dplyr::select(-"X")
gene_list_PBS_vs_PBS_MTX_D3_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS_vs_PBS_D3_D22.csv") %>% dplyr::select(-"X")


# Load diff_genes : PBS vs PBS (OCP)
gene_list_PBS_vs_PBS_OCP_D10_D22 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS(OCP)_vs_PBS(OCP)_D10_D22.csv") %>% dplyr::select(-"X") 
gene_list_PBS_vs_PBS_OCP_D3_D10 <- read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS(OCP)_vs_PBS(OCP)_D3_D10.csv") %>% dplyr::select(-"X") 
gene_list_PBS_vs_PBS_OCP_D3_D22 <-read.csv("/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_PBS(OCP)_vs_PBS(OCP)_D3_D22.csv") %>% dplyr::select(-"X")

```





```{r}
  
# Overlapping differentially expressed genes : MTX vs PBS  
get_overlapping_diff_genes(gene_list_MTX_vs_PBS_D3, gene_list_MTX_vs_PBS_D10, "/data/users/tlassale/MODICED_link/common_diff_genes_MTX(D3)_vs_PBS(D3)_vs_MTX(D10)_vs_PBS(D10).csv")
get_venn_diagram_overlap(gene_list_MTX_vs_PBS_D3, gene_list_MTX_vs_PBS_D10, gene_list_MTX_vs_PBS_D22, "MTX vs PBS", "VennDiagram_MTX_vs_PBS.png")
 

# Overlapping differentially expressed genes : Oxa vs PBS  
get_overlapping_diff_genes(gene_list_Oxa_vs_PBS_D3, gene_list_Oxa_vs_PBS_D10, "/data/users/tlassale/MODICED_link/common_diff_genes_Oxa(D3)_vs_PBS(D3)_vs_Oxa(D10)_vs_PBS(D10).csv")
get_overlapping_diff_genes(gene_list_Oxa_vs_PBS_D10, gene_list_Oxa_vs_PBS_D22, "/data/users/tlassale/MODICED_link/common_diff_genes_Oxa(D10)_vs_PBS(D10)_vs_Oxa(D22)_vs_PBS(D22).png")
get_overlapping_diff_genes(gene_list_Oxa_vs_PBS_D3, gene_list_Oxa_vs_PBS_D22, "/data/users/tlassale/MODICED_link/common_diff_genes_Oxa(D3)_vs_PBS(D3)_vs_Oxa(D22)_vs_PBS(D22).csv")
get_venn_diagram_overlap(gene_list_Oxa_vs_PBS_D3, gene_list_Oxa_vs_PBS_D10, gene_list_Oxa_vs_PBS_D22, "Oxa vs PBS", "VennDiagram_Oxa_vs_PBS.png")
 

# Overlapping differentially expressed genes : Cis vs PBS  
get_overlapping_diff_genes(gene_list_Cis_vs_PBS_D3, gene_list_Cis_vs_PBS_D10, "/data/users/tlassale/MODICED_link/common_diff_genes_Cis(D3)_vs_PBS(D3)_vs_Cis(D10)_vs_PBS(D10).csv")
get_overlapping_diff_genes(gene_list_Cis_vs_PBS_D10, gene_list_Cis_vs_PBS_D22, "/data/users/tlassale/MODICED_link/common_diff_genes_Cis(D10)_vs_PBS(D10)_vs_Cis(D22)_vs_PBS(D22).png")
get_overlapping_diff_genes(gene_list_Cis_vs_PBS_D3, gene_list_Cis_vs_PBS_D22, "/data/users/tlassale/MODICED_link/common_diff_genes_Cis(D3)_vs_PBS(D3)_vs_Cis(D22)_vs_PBS(D22).csv")
get_venn_diagram_overlap(gene_list_Cis_vs_PBS_D3, gene_list_Cis_vs_PBS_D10, gene_list_Cis_vs_PBS_D22, "Cis vs PBS", "VennDiagram_Cis_vs_PBS.png")
 

# Overlapping differentially expressed genes : MTX vs MTX  
get_overlapping_diff_genes(gene_list_MTX_vs_MTX_D22_D10, gene_list_MTX_vs_MTX_D10_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_MTX(D22)_vs_MTX(D10)_vs_MTX(D10)_vs_PBS(D3).csv")
get_overlapping_diff_genes(gene_list_MTX_vs_MTX_D22_D10, gene_list_MTX_vs_MTX_D22_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_MTX(D22)_vs_MTX(D10)_vs_MTX(D22)_vs_PBS(D3).png")

get_venn_diagram_overlap(gene_list_MTX_vs_MTX_D22_D10, gene_list_MTX_vs_MTX_D10_D3, gene_list_MTX_vs_MTX_D22_D3, "MTX vs MTX", "VennDiagram_MTX_vs_MTX.png")

# Overlapping differentially expressed genes : Oxa vs Oxa  
get_overlapping_diff_genes(gene_list_Oxa_vs_Oxa_D22_D10, gene_list_Oxa_vs_Oxa_D10_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_Oxa(D22)_vs_Oxa(D10)_vs_Oxa(D10)_vs_PBS(D3).csv")
get_overlapping_diff_genes(gene_list_Oxa_vs_Oxa_D22_D10, gene_list_Oxa_vs_Oxa_D22_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_Oxa(D22)_vs_Oxa(D10)_vs_Oxa(D22)_vs_PBS(D3).png")

get_venn_diagram_overlap(gene_list_Oxa_vs_Oxa_D22_D10, gene_list_Oxa_vs_Oxa_D10_D3, gene_list_Oxa_vs_Oxa_D22_D3, "Oxa vs Oxa", "VennDiagram_Oxa_vs_Oxa.png")
 

# Overlapping differentially expressed genes : Cis vs Cis  
get_overlapping_diff_genes(gene_list_Cis_vs_Cis_D22_D10, gene_list_Cis_vs_Cis_D10_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_Cis(D22)_vs_Cis(D10)_vs_Cis(D10)_vs_PBS(D3).csv")
get_overlapping_diff_genes(gene_list_Cis_vs_Cis_D22_D10, gene_list_Cis_vs_Cis_D22_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_Cis(D22)_vs_Cis(D10)_vs_Cis(D22)_vs_PBS(D3).csv")

get_venn_diagram_overlap(gene_list_Cis_vs_Cis_D22_D10, gene_list_Cis_vs_Cis_D10_D3, gene_list_Cis_vs_Cis_D22_D3, "Cis vs Cis", "VennDiagram_Cis_vs_Cis.png")


# Overlapping differentially expressed genes : PBS_MTX vs PBS_MTX
get_overlapping_diff_genes(gene_list_PBS_vs_PBS_MTX_D22_D10, gene_list_PBS_vs_PBS_MTX_D10_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_PBS(D22)_vs_PBS(D10)_vs_PBS(D10)_vs_PBS(D3)_MTX.csv")
get_overlapping_diff_genes(gene_list_PBS_vs_PBS_MTX_D22_D10, gene_list_PBS_vs_PBS_MTX_D22_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_PBS(D22)_vs_PBS(D10)_vs_PBS(D22)_vs_PBS(D3)_MTX.png")

get_venn_diagram_overlap(gene_list_PBS_vs_PBS_MTX_D22_D10, gene_list_PBS_vs_PBS_MTX_D10_D3, gene_list_PBS_vs_PBS_MTX_D22_D3, "PBS_MTX vs PBS_MTX", "VennDiagram_PBS_vs_PBS_MTX.png")


# Overlapping differentially expressed genes : PBS_OCP vs PBS_OCP
get_overlapping_diff_genes(gene_list_PBS_vs_PBS_OCP_D22_D10, gene_list_PBS_vs_PBS_OCP_D10_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_PBS(D22)_vs_PBS(D10)_vs_PBS(D10)_vs_PBS(D3)_OCP.csv")
get_overlapping_diff_genes(gene_list_PBS_vs_PBS_OCP_D22_D10, gene_list_PBS_vs_PBS_OCP_D22_D3, "/data/users/tlassale/MODICED_link/common_diff_genes_PBS(D22)_vs_PBS(D10)_vs_PBS(D22)_vs_PBS(D3)_OCP.csv")

get_venn_diagram_overlap(gene_list_PBS_vs_PBS_OCP_D22_D10, gene_list_PBS_vs_PBS_OCP_D10_D3, gene_list_PBS_vs_PBS_OCP_D22_D3, "PBS_OCP vs PBS_OCP", "VennDiagram_PBS_vs_PBS_OCP.png")


  
```






# **4. Gene set enrichment analysis** 

## **4.1. GO enrichment analysis**


```{r}

# GO dotplot function
enrich_plot_go <- function(test){
if (nrow(test) >= 20){
  ggplot(data = test[1:20,], 
       mapping = aes(x = prop_overlap,
                     y = reorder(Biological_Processes,prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("Biological Processes") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
else {
    ggplot(data = test, 
       mapping = aes(x = prop_overlap,
                     y = reorder(Biological_Processes,prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("Biological Processes") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
}
```


```{r}
# Get GO final results (Dotplot)
get_go <- function(filename, filename_2, filename_3, filename_4, filename_5, filename_6, filename_7){
  
  # Get diff genes
  diff <- read.csv(filename)
  diff_up <- diff %>% filter(group == "up")
  diff_down <- diff %>% filter(group == "down")
  
  
  # Get gene symbol
  all_genes_symbol <- diff$genes
  diff_up_symbol <- diff_up$genes
  diff_down_symbol <- diff_down$genes
  
  # Retrieve ENSEMBL and ENTREZID
  res_bitr_all <- bitr(all_genes_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_up <- bitr(diff_up_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_down <- bitr(diff_down_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  
  # Get ENTREZ ID for list 
  res_bitr_all_list <- list(genes = res_bitr_all$ENTREZID)
  res_bitr_up_list <- list(genes = res_bitr_up$ENTREZID)
  res_bitr_down_list <- list(genes = res_bitr_down$ENTREZID)
  
  # Run GO enrichment analysis
  go_analysis_res_bitr_all_df <- enrichGO(res_bitr_all_list$genes, ont="BP", OrgDb = "org.Mm.eg.db", readable = TRUE)
  go_analysis_res_bitr_up_df <- enrichGO(res_bitr_up_list$genes, ont="BP", OrgDb = "org.Mm.eg.db", readable = TRUE)
  go_analysis_res_bitr_down_df <- enrichGO(res_bitr_down_list$genes, ont="BP", OrgDb = "org.Mm.eg.db", readable = TRUE)
  
  
  # Compute proportion of genes in the GO term that overlaps with the DEGs
  go_analysis_res_bitr_all_df_not_redundant <- simplify(go_analysis_res_bitr_all_df, cutoff=0.7, by="p.adjust", select_fun=min)
  go_analysis_res_bitr_all_df_not_redundant <- go_analysis_res_bitr_all_df_not_redundant %>% dplyr::mutate(prop_overlap  = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),
                                                     Biological_Processes = Description)
  
  
  go_analysis_res_bitr_up_df_not_redundant <- simplify(go_analysis_res_bitr_up_df, cutoff=0.7, by="p.adjust", select_fun=min)
  go_analysis_res_bitr_up_df_not_redundant <- go_analysis_res_bitr_up_df_not_redundant %>% dplyr::mutate(prop_overlap  = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),
                                                     Biological_Processes = Description)
  
  
  go_analysis_res_bitr_down_df_not_redundant <- simplify(go_analysis_res_bitr_down_df, cutoff=0.7, by="p.adjust", select_fun=min)
  go_analysis_res_bitr_down_df_not_redundant <- go_analysis_res_bitr_down_df_not_redundant %>% dplyr::mutate(prop_overlap  = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),
                                                     Biological_Processes = Description)
  
  
  
  # Convert prop overlap in percentage
  go_analysis_res_bitr_all_df_not_redundant <- go_analysis_res_bitr_all_df_not_redundant %>% mutate(prop_overlap = prop_overlap * 100)
  
  go_analysis_res_bitr_up_df_not_redundant <- go_analysis_res_bitr_up_df_not_redundant %>% mutate(prop_overlap = prop_overlap * 100)
  
  go_analysis_res_bitr_down_df_not_redundant <- go_analysis_res_bitr_down_df_not_redundant %>% mutate(prop_overlap = prop_overlap * 100)
  

  # Rank GO terms by prop of overlap
  ordered_go_analysis_res_bitr_all_df <- go_analysis_res_bitr_all_df_not_redundant[order(go_analysis_res_bitr_all_df_not_redundant$prop_overlap, decreasing = TRUE),]
  write.csv(ordered_go_analysis_res_bitr_all_df, filename_5)
  
  ordered_go_analysis_res_bitr_up_df <- go_analysis_res_bitr_up_df_not_redundant[order(go_analysis_res_bitr_up_df_not_redundant$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_go_analysis_res_bitr_up_df, filename_6)
  
  ordered_go_analysis_res_bitr_down_df <- go_analysis_res_bitr_down_df_not_redundant[order(go_analysis_res_bitr_down_df_not_redundant$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_go_analysis_res_bitr_down_df, filename_7)
  

  # Dotplot and save it
  enrich_plot_go(ordered_go_analysis_res_bitr_all_df)
  ggsave(filename_2, width=12, height=4)
  enrich_plot_go(ordered_go_analysis_res_bitr_up_df)
  ggsave(filename_3, width=12, height=4)
  enrich_plot_go(ordered_go_analysis_res_bitr_down_df)
  ggsave(filename_4, width=12, height=4)

}

```

```{r}
overlap_intersect_go_by_2 <- function(filename_1, filename_2, filename_3){
  enriched_go_terms_1 <- read.csv(filename_1)
  enriched_go_terms_2 <- read.csv(filename_2)
  intersect_go_terms <- intersect(enriched_go_terms_1$Biological_Processes, enriched_go_terms_2$Biological_Processes)
  write.csv(overlap_intersect_go_by_2, filename_3)
}

overlap_intersect_go_by_3 <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_go_terms_1 <- read.csv(filename_1)
  enriched_go_terms_2 <- read.csv(filename_2)
  enriched_go_terms_3 <- read.csv(filename_3)
  intersect_go_terms <- intersect(intersect(enriched_go_terms_1$Biological_Processes, enriched_go_terms_2$Biological_Processes), enriched_go_terms_3$Biological_Processes)
  write.csv(intersect_go_terms, filename_4)
  list_intersect_go_terms <- list(Oxa_vs_PBS = enriched_go_terms_1$Biological_Processes, Cis_vs_PBS = enriched_go_terms_2$Biological_Processes, MTX_vs_PBS = enriched_go_terms_3$Biological_Processes)
  ggvenn(list_intersect_go_terms, stroke_size = 0.5)
  ggsave(filename_5)
}

overlap_intersect_go_by_3_by_treatment <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_go_terms_1 <- read.csv(filename_1)
  enriched_go_terms_2 <- read.csv(filename_2)
  enriched_go_terms_3 <- read.csv(filename_3)
  intersect_go_terms <- intersect(intersect(enriched_go_terms_1$Biological_Processes, enriched_go_terms_2$Biological_Processes), enriched_go_terms_3$Biological_Processes)
  write.csv(intersect_go_terms, filename_4)
  list_intersect_go_terms <- list(up_reg_genes_BP = enriched_go_terms_1$Biological_Processes, down_reg_genes_BP = enriched_go_terms_2$Biological_Processes, all_genes_BP = enriched_go_terms_3$Biological_Processes)
  ggvenn(list_intersect_go_terms, stroke_size = 0.5)
  ggsave(filename_5)
}
```



### D3

```{r}
get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): MTX(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): MTX(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): MTX(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_down_genes.csv")

get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): Oxa(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): Oxa(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): Oxa(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_down_genes.csv")

get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): Cis(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): Cis(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): Cis(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_down_genes.csv")
```


```{r}
# Overlapping enriched GO term on Day 3 between Oxa and Cis (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Cis (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Oxa (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX, Oxa and Cis (all genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.png")

```


```{r}
# Overlapping enriched GO term on Day 3 between Oxa and Cis (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Cis (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Oxa (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX, Oxa and Cis (up genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.png")
```


```{r}
# Overlapping enriched GO term on Day 3 between Oxa and Cis (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Cis (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched GO term on Day 3 between MTX and Oxa (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_down_genes.csv")


# Overlapping enriched GO term on Day 3 between MTX, Oxa and Cis (down genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.png")
```


```{r}
# Overlapping enriched GO term between up, down and all the genes for MTX
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.png")



# Overlapping enriched GO term between up, down and all the genes for OXA
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.png")


# Overlapping enriched GO term between up, down and all the genes for CIS
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.png")
```





### D10

```{r}
get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): MTX(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): MTX(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): MTX(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_down_genes.csv")

get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): Oxa(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): Oxa(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): Oxa(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_down_genes.csv")

get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): Cis(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): Cis(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): Cis(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_down_genes.csv")
```


```{r}
# Overlapping enriched GO term on Day 10 between Oxa and Cis (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Cis (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Oxa (all genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX, Oxa and Cis (all genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.png")

```


```{r}
# Overlapping enriched GO term on Day 10 between Oxa and Cis (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Cis (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Oxa (up genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX, Oxa and Cis (up genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.png")
```


```{r}
# Overlapping enriched GO term on Day 10 between Oxa and Cis (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Cis (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched GO term on Day 10 between MTX and Oxa (down genes)
overlap_intersect_go_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_down_genes.csv")


# Overlapping enriched GO term on Day 10 between MTX, Oxa and Cis (down genes)
overlap_intersect_go_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.png")
```


```{r}
# Overlapping enriched GO term between up, down and all the genes for MTX
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.png")



# Overlapping enriched GO term between up, down and all the genes for OXA
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.png")


# Overlapping enriched GO term between up, down and all the genes for CIS
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.png")
```


### D22

```{r}

get_go(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D22.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (all genes): Oxa(D22) vs PBS(D22).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (up reg genes): Oxa(D22) vs PBS(D22).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/GO enrichment (down reg genes): Oxa(D22) vs PBS(D22).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_down_genes.csv")

```

```{r}
# Overlapping enriched GO term between up, down and all the genes for OXA
overlap_intersect_go_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_go_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_go_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.png")
```




## **4.2. Reactome enrichment analysis**

```{r}
# Reactome dotplot function
enrich_plot_reactome <- function(test){
if (nrow(test) >= 20){
  ggplot(data = test[1:20,], 
       mapping = aes(x = prop_overlap,
                     y = reorder(Reactome_Pathway, prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("Reactome Pathway") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
else {
    ggplot(data = test, 
       mapping = aes(x = prop_overlap,
                     y = reorder(Reactome_Pathway,prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("Reactome Pathway") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
}
```


```{r}
# Get reactome final results (Dotplot)
get_reactome <- function(filename, filename_2, filename_3, filename_4, filename_5, filename_6, filename_7){
  
  # Get diff genes
  diff <- read.csv(filename)
  diff_up <- diff %>% filter(group == "up")
  diff_down <- diff %>% filter(group == "down")
  
  
  # Get gene symbol
  all_genes_symbol <- diff$genes
  diff_up_symbol <- diff_up$genes
  diff_down_symbol <- diff_down$genes
  
  # Retrieve ENSEMBL and ENTREZID
  res_bitr_all <- bitr(all_genes_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_up <- bitr(diff_up_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_down <- bitr(diff_down_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  
  # Get ENTREZ ID for list 
  res_bitr_all_list <- list(genes = res_bitr_all$ENTREZID)
  res_bitr_up_list <- list(genes = res_bitr_up$ENTREZID)
  res_bitr_down_list <- list(genes = res_bitr_down$ENTREZID)
  
  # Run reactome enrichment analysis
  reactome_analysis_res_bitr_all_df <- enrichPathway(res_bitr_all_list$genes, organism="mouse")
  reactome_analysis_res_bitr_up_df <- enrichPathway(res_bitr_up_list$genes, organism="mouse")
  reactome_analysis_res_bitr_down_df <- enrichPathway(res_bitr_down_list$genes, organism="mouse")
  
# Get prop of overlap (+ convert in percentage)
reactome_analysis_res_bitr_all_df <- reactome_analysis_res_bitr_all_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Reactome_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100) 


reactome_analysis_res_bitr_up_df <- reactome_analysis_res_bitr_up_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Reactome_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100)
  


reactome_analysis_res_bitr_down_df <- reactome_analysis_res_bitr_down_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Reactome_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100)


# Rank reactome terms by prop of overlap
  ordered_reactome_analysis_res_bitr_all_df <- reactome_analysis_res_bitr_all_df[order(reactome_analysis_res_bitr_all_df$prop_overlap, decreasing = TRUE),]
  write.csv(ordered_reactome_analysis_res_bitr_all_df, filename_5)
  
  ordered_reactome_analysis_res_bitr_up_df <- reactome_analysis_res_bitr_up_df[order(reactome_analysis_res_bitr_up_df$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_reactome_analysis_res_bitr_up_df, filename_6)
  
  ordered_reactome_analysis_res_bitr_down_df <- reactome_analysis_res_bitr_down_df[order(reactome_analysis_res_bitr_down_df$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_reactome_analysis_res_bitr_down_df, filename_7)
  

  # Dotplot and save it
  enrich_plot_reactome(ordered_reactome_analysis_res_bitr_all_df)
  ggsave(filename_2, width=12, height=4)
  enrich_plot_reactome(ordered_reactome_analysis_res_bitr_up_df)
  ggsave(filename_3, width=12, height=4)
  enrich_plot_reactome(ordered_reactome_analysis_res_bitr_down_df)
  ggsave(filename_4, width=12, height=4)

}

```

```{r}
overlap_intersect_reactome_by_2 <- function(filename_1, filename_2, filename_3){
  enriched_reactome_pathways_1 <- read.csv(filename_1)
  enriched_reactome_pathways_2 <- read.csv(filename_2)
  intersect_reactome_pathways <- intersect(enriched_reactome_pathways_1$Reactome_Pathway, enriched_reactome_pathways_2$Reactome_Pathway)
  write.csv(overlap_intersect_reactome_by_2, filename_3)
}

overlap_intersect_reactome_by_3 <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_reactome_pathways_1 <- read.csv(filename_1)
  enriched_reactome_pathways_2 <- read.csv(filename_2)
  enriched_reactome_pathways_3 <- read.csv(filename_3)
  intersect_reactome_pathways <- intersect(intersect(enriched_reactome_pathways_1$Reactome_Pathway, enriched_reactome_pathways_2$Reactome_Pathway), enriched_reactome_pathways_3$Reactome_Pathway)
  write.csv(intersect_reactome_pathways, filename_4)
  list_intersect_reactome_pathways <- list(Oxa_vs_PBS = enriched_reactome_pathways_1$Reactome_Pathway, Cis_vs_PBS = enriched_reactome_pathways_2$Reactome_Pathway, MTX_vs_PBS = enriched_reactome_pathways_3$Reactome_Pathway)
  ggvenn(list_intersect_reactome_pathways, stroke_size = 0.5)
  ggsave(filename_5)
}

overlap_intersect_reactome_by_3_by_treatment <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_reactome_terms_1 <- read.csv(filename_1)
  enriched_reactome_terms_2 <- read.csv(filename_2)
  enriched_reactome_terms_3 <- read.csv(filename_3)
  intersect_reactome_terms <- intersect(intersect(enriched_reactome_terms_1$Reactome_Pathway, enriched_reactome_terms_2$Reactome_Pathway), enriched_reactome_terms_3$Reactome_Pathway)
  write.csv(intersect_reactome_terms, filename_4)
  list_intersect_reactome_terms <- list(up_reg_genes_reactome = enriched_reactome_terms_1$Reactome_Pathway, down_reg_genes_reactome = enriched_reactome_terms_2$Reactome_Pathway, all_genes_reactome = enriched_reactome_terms_3$Reactome_Pathway)
  ggvenn(list_intersect_reactome_terms, stroke_size = 0.5)
  ggsave(filename_5)
}
```



### D3

```{r}
get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): MTX(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): MTX(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): MTX(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_down_genes.csv")

get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): Oxa(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): Oxa(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): Oxa(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_down_genes.csv")

get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): Cis(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): Cis(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): Cis(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_down_genes.csv")
```


```{r}
# Overlapping enriched reactome term on Day 3 between Oxa and Cis (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX and Cis (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX and Oxa (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX, Oxa and Cis (all genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_Reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_Reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.png")

```


```{r}
# Overlapping enriched reactome term on Day 3 between Oxa and Cis (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX and Cis (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX and Oxa (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched reactome term on Day 3 between MTX, Oxa and Cis (up genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_Reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_Reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.png")
```


```{r}
# Overlapping enriched reactome pathway on Day 3 between Oxa and Cis (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched reactome pathway on Day 3 between MTX and Cis (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched reactome pathway on Day 3 between MTX and Oxa (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_down_genes.csv")


# Overlapping enriched reactome pathway on Day 3 between MTX, Oxa and Cis (down genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.png")
```


```{r}
# Overlapping enriched reactome pathway between up, down and all the genes for MTX
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.png")



# Overlapping enriched reactome pathway between up, down and all the genes for OXA
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.png")


# Overlapping enriched reactome pathway between up, down and all the genes for CIS
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.png")
```


### D10

```{r}
get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): MTX(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): MTX(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): MTX(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_down_genes.csv")

get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): Oxa(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): Oxa(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): Oxa(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_down_genes.csv")

get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): Cis(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): Cis(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): Cis(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_down_genes.csv")
```


```{r}
# Overlapping enriched reactome term on Day 10 between Oxa and Cis (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched reactome term on Day 10 between MTX and Cis (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched reactome term on Day 10 between MTX and Oxa (all genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched reactome term on Day 10 between MTX, Oxa and Cis (all genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.png")

```


```{r}
# Overlapping enriched reactome pathway on Day 10 between Oxa and Cis (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched reactome pathway on Day 10 between MTX and Cis (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched reactome pathway on Day 10 between MTX and Oxa (up genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched reactome pathway on Day 10 between MTX, Oxa and Cis (up genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.png")
```


```{r}
# Overlapping enriched reactome pathway on Day 10 between Oxa and Cis (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched reactome pathway on Day 10 between MTX and Cis (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched reactome pathway on Day 10 between MTX and Oxa (down genes)
overlap_intersect_reactome_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_down_genes.csv")


# Overlapping enriched reactome pathway on Day 10 between MTX, Oxa and Cis (down genes)
overlap_intersect_reactome_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.png")
```


```{r}
# Overlapping enriched reactome pathway between up, down and all the genes for MTX
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.png")



# Overlapping enriched reactome pathway between up, down and all the genes for OXA
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.png")


# Overlapping enriched reactome pathway between up, down and all the genes for CIS
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.png")
```



### D22

```{r}

get_reactome(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D22.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (all genes): Oxa(D22) vs PBS(D22).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (up reg genes): Oxa(D22) vs PBS(D22).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Reactome enrichment (down reg genes): Oxa(D22) vs PBS(D22).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_down_genes.csv")

```

```{r}
# Overlapping enriched GO term between up, down and all the genes for OXA
overlap_intersect_reactome_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_Reactome_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_reactome_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.png")
```


## **4.3. Kegg enrichment analysis**

```{r}
# kegg dotplot function
enrich_plot_kegg <- function(test){
if (nrow(test) >= 20){
  ggplot(data = test[1:20,], 
       mapping = aes(x = prop_overlap,
                     y = reorder(Kegg_Pathway, prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("kegg Pathway") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
else {
    ggplot(data = test, 
       mapping = aes(x = prop_overlap,
                     y = reorder(Kegg_Pathway,prop_overlap),
                     color = p.adjust,
                     size = prop_overlap
                     )
       ) + 
geom_point() + 
scale_color_gradient(low = "red", high = "blue") + 
theme_bw() +
xlab("Proportion of Overlap (in %)") +
ylab("Kegg Pathway") +
theme(axis.text= element_text(face = "bold")) +
theme(axis.title.x = element_text(size = 9))
}
}
```


```{r}
# Get kegg final results (Dotplot)
get_kegg <- function(filename, filename_2, filename_3, filename_4, filename_5, filename_6, filename_7){
  
  # Get diff genes
  diff <- read.csv(filename)
  diff_up <- diff %>% filter(group == "up")
  diff_down <- diff %>% filter(group == "down")
  
  
  # Get gene symbol
  all_genes_symbol <- diff$genes
  diff_up_symbol <- diff_up$genes
  diff_down_symbol <- diff_down$genes
  
  # Retrieve ENSEMBL and ENTREZID
  res_bitr_all <- bitr(all_genes_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_up <- bitr(diff_up_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  res_bitr_down <- bitr(diff_down_symbol, fromType = "SYMBOL", toType = "ENTREZID", OrgDb = "org.Mm.eg.db")
  
  # Get ENTREZ ID for list 
  res_bitr_all_list <- list(genes = res_bitr_all$ENTREZID)
  res_bitr_up_list <- list(genes = res_bitr_up$ENTREZID)
  res_bitr_down_list <- list(genes = res_bitr_down$ENTREZID)
  
  # Run kegg enrichment analysis
  kegg_analysis_res_bitr_all_df <- enrichKEGG(res_bitr_all_list$genes, organism="mmu")
  kegg_analysis_res_bitr_up_df <- enrichKEGG(res_bitr_up_list$genes, organism = "mmu")
  kegg_analysis_res_bitr_down_df <- enrichKEGG(res_bitr_down_list$genes, organism = "mmu")
  
  

  # Get prop of overlap (+ convert in percentage)
kegg_analysis_res_bitr_all_df <- kegg_analysis_res_bitr_all_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Kegg_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100) 


kegg_analysis_res_bitr_up_df <- kegg_analysis_res_bitr_up_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Kegg_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100)
  


kegg_analysis_res_bitr_down_df <- kegg_analysis_res_bitr_down_df %>%
                                              as.data.frame() %>% 
                                              mutate(prop_overlap = as.integer(str_remove(GeneRatio, "/.*")) / as.integer(str_remove(BgRatio, "/.*")),                                                   Kegg_Pathway = Description) %>% mutate(prop_overlap = prop_overlap * 100)
  

  # Rank kegg terms by prop of overlap
  ordered_kegg_analysis_res_bitr_all_df <- kegg_analysis_res_bitr_all_df[order(kegg_analysis_res_bitr_all_df$prop_overlap, decreasing = TRUE),]
  write.csv(ordered_kegg_analysis_res_bitr_all_df, filename_5)
  
  ordered_kegg_analysis_res_bitr_up_df <- kegg_analysis_res_bitr_up_df[order(kegg_analysis_res_bitr_up_df$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_kegg_analysis_res_bitr_up_df, filename_6)
  
  ordered_kegg_analysis_res_bitr_down_df <- kegg_analysis_res_bitr_down_df[order(kegg_analysis_res_bitr_down_df$prop_overlap, decreasing = TRUE),] 
  write.csv(ordered_kegg_analysis_res_bitr_down_df, filename_7)
  

  # Dotplot and save it
  enrich_plot_kegg(ordered_kegg_analysis_res_bitr_all_df)
  ggsave(filename_2, width=12, height=4)
  enrich_plot_kegg(ordered_kegg_analysis_res_bitr_up_df)
  ggsave(filename_3, width=12, height=4)
  enrich_plot_kegg(ordered_kegg_analysis_res_bitr_down_df)
  ggsave(filename_4, width=12, height=4)

}

```

```{r}
overlap_intersect_kegg_by_2 <- function(filename_1, filename_2, filename_3){
  enriched_kegg_pathways_1 <- read.csv(filename_1)
  enriched_kegg_pathways_2 <- read.csv(filename_2)
  intersect_kegg_pathways <- intersect(enriched_kegg_pathways_1$kegg_Pathway, enriched_kegg_pathways_2$kegg_Pathway)
  write.csv(overlap_intersect_kegg_by_2, filename_3)
}

overlap_intersect_kegg_by_3 <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_kegg_pathways_1 <- read.csv(filename_1)
  enriched_kegg_pathways_2 <- read.csv(filename_2)
  enriched_kegg_pathways_3 <- read.csv(filename_3)
  intersect_kegg_pathways <- intersect(intersect(enriched_kegg_pathways_1$Kegg_Pathway, enriched_kegg_pathways_2$Kegg_Pathway), enriched_kegg_pathways_3$Kegg_Pathway)
  write.csv(intersect_kegg_pathways, filename_4)
  list_intersect_kegg_pathways <- list(Oxa_vs_PBS = enriched_kegg_pathways_1$Kegg_Pathway, Cis_vs_PBS = enriched_kegg_pathways_2$Kegg_Pathway, MTX_vs_PBS = enriched_kegg_pathways_3$Kegg_Pathway)
  ggvenn(list_intersect_kegg_pathways, stroke_size = 0.5)
  ggsave(filename_5)
}


overlap_intersect_kegg_by_3_by_treatment <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_kegg_terms_1 <- read.csv(filename_1)
  enriched_kegg_terms_2 <- read.csv(filename_2)
  enriched_kegg_terms_3 <- read.csv(filename_3)
  intersect_kegg_terms <- intersect(intersect(enriched_kegg_terms_1$Kegg_Pathway, enriched_kegg_terms_2$Kegg_Pathway), enriched_kegg_terms_3$Kegg_Pathway)
  write.csv(intersect_kegg_terms, filename_4)
  list_intersect_kegg_terms <- list(up_reg_genes_kegg = enriched_kegg_terms_1$Kegg_Pathway, down_reg_genes_kegg = enriched_kegg_terms_2$Kegg_Pathway, all_genes_kegg = enriched_kegg_terms_3$Kegg_Pathway)
  ggvenn(list_intersect_kegg_terms, stroke_size = 0.5)
  ggsave(filename_5)
}

overlap_intersect_kegg_by_3_by_treatment <- function(filename_1, filename_2, filename_3, filename_4, filename_5){
  enriched_kegg_terms_1 <- read.csv(filename_1)
  enriched_kegg_terms_2 <- read.csv(filename_2)
  enriched_kegg_terms_3 <- read.csv(filename_3)
  intersect_kegg_terms <- intersect(intersect(enriched_kegg_terms_1$Kegg_Pathway, enriched_kegg_terms_2$Kegg_Pathway), enriched_kegg_terms_3$Kegg_Pathway)
  write.csv(intersect_kegg_terms, filename_4)
  list_intersect_kegg_terms <- list(up_reg_genes_kegg = enriched_kegg_terms_1$Kegg_Pathway, down_reg_genes_kegg = enriched_kegg_terms_2$Kegg_Pathway, all_genes_kegg = enriched_kegg_terms_3$Kegg_Pathway)
  ggvenn(list_intersect_kegg_terms, stroke_size = 0.5)
  ggsave(filename_5)
}
```


### D3

```{r}
get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): MTX(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): MTX(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): MTX(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_down_genes.csv")

get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): Oxa(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): Oxa(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): Oxa(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_down_genes.csv")

get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D3.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): Cis(D3) vs PBS(D3).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): Cis(D3) vs PBS(D3).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): Cis(D3) vs PBS(D3).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_down_genes.csv")
```


```{r}
# Overlapping enriched kegg term on Day 3 between Oxa and Cis (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Cis (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Oxa (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_all_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX, Oxa and Cis (all genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_all_genes.png")

```


```{r}
# Overlapping enriched kegg term on Day 3 between Oxa and Cis (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Cis (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Oxa (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_up_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX, Oxa and Cis (up genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_up_genes.png")
```


```{r}
# Overlapping enriched kegg term on Day 3 between Oxa and Cis (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Cis (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv")

# Overlapping enriched kegg term on Day 3 between MTX and Oxa (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_down_genes.csv")


# Overlapping enriched kegg term on Day 3 between MTX, Oxa and Cis (down genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_vs_Oxa(D3)_vs_PBS(D3)_vs_Cis(D3)_vs_PBS(D3)_down_genes.png")
```

```{r}
# Overlapping enriched kegg pathway between up, down and all the genes for MTX
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D3)_vs_PBS(D3)_all_up_down_genes.png")



# Overlapping enriched kegg pathway between up, down and all the genes for OXA
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D3)_vs_PBS(D3)_all_up_down_genes.png")


# Overlapping enriched kegg pathway between up, down and all the genes for CIS
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D3) vs PBS(D3)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_CIS(D3)_vs_PBS(D3)_all_up_down_genes.png")
```



### D10

```{r}
get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_MTX_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): MTX(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): MTX(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): MTX(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_down_genes.csv")

get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): Oxa(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): Oxa(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): Oxa(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_down_genes.csv")

get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_CIS_vs_PBS_D10.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): Cis(D10) vs PBS(D10).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): Cis(D10) vs PBS(D10).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): Cis(D10) vs PBS(D10).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_down_genes.csv")
```


```{r}
# Overlapping enriched kegg term on Day 10 between Oxa and Cis (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Cis (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Oxa (all genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_all_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX, Oxa and Cis (all genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_all_genes.png")

```


```{r}
# Overlapping enriched kegg term on Day 10 between Oxa and Cis (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Cis (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Oxa (up genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_up_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX, Oxa and Cis (up genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_up_genes.png")
```


```{r}
# Overlapping enriched kegg term on Day 10 between Oxa and Cis (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Cis (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX and Oxa (down genes)
overlap_intersect_kegg_by_2(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_down_genes.csv")

# Overlapping enriched kegg term on Day 10 between MTX, Oxa and Cis (down genes)
overlap_intersect_kegg_by_3(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_vs_Oxa(D10)_vs_PBS(D10)_vs_Cis(D10)_vs_PBS(D10)_down_genes.png")

```


```{r}
# Overlapping enriched kegg pathway between up, down and all the genes for MTX
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_MTX(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_MTX(D10)_vs_PBS(D10)_all_up_down_genes.png")



# Overlapping enriched kegg pathway between up, down and all the genes for OXA
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D10)_vs_PBS(D10)_all_up_down_genes.png")


# Overlapping enriched kegg pathway between up, down and all the genes for CIS
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Cis(D10) vs PBS(D10)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_CIS(D10)_vs_PBS(D10)_all_up_down_genes.png")
```


### D22

```{r}


get_kegg(filename = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/gene_list_OXA_vs_PBS_D22.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (all genes): Oxa(D22) vs PBS(D22).png", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (up reg genes): Oxa(D22) vs PBS(D22).png", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/kegg enrichment (down reg genes): Oxa(D22) vs PBS(D22).png", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_6 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_7 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_down_genes.csv")

```

```{r}
# Overlapping enriched GO term between up, down and all the genes for OXA
overlap_intersect_kegg_by_3_by_treatment(filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_up_genes.csv", filename_2 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_down_genes.csv", filename_3 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/enriched_kegg_terms_Oxa(D22) vs PBS(D22)_all_genes.csv", filename_4 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.csv", filename_5 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/overlapping_kegg_terms_OXA(D22)_vs_PBS(D22)_all_up_down_genes.png")
```



# **5. Immune Deconvolution analysis** 


```{r}

tablecounts_tpm <- read.csv("/data/kdi_prod/.kdi/project_workspace_0/1929/acl/01.00/results/counts/tablecounts_tpm.csv", header=TRUE) %>%                         as.data.frame() %>%
                 dplyr::rename("gene_id"  ="X") %>%
                 inner_join(annot_gene[,c("gene_id","gene_name")], by="gene_id") %>%
                 relocate(gene_name, .after = gene_id) %>%
                 dplyr::select(-gene_id) %>%
                 magrittr::set_rownames(make.unique(.$gene_name)) %>%
                 dplyr::select(-gene_name) %>%
                 as.matrix()
                

# Load human and mouse ensembl datasets annotation
human_mart <- useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl", host = "https://dec2021.archive.ensembl.org")

musculus_mart <- useMart(biomart="ensembl", dataset="mmusculus_gene_ensembl", host = "https://dec2021.archive.ensembl.org")

# Create table of mouse and human 
humanGeneName_mouseGeneName <- getLDS(attributes = "hgnc_symbol",
                                      mart       = human_mart,
                                      attributesL= "mgi_symbol",
                                      martL      = musculus_mart,
                                      uniqueRows = TRUE
                                      )


# Convert mouse gene name to human gene name
tablecounts_tpm_plus <- merge(x = tablecounts_tpm,
                              y = humanGeneName_mouseGeneName,
                              by.x = 0,
                              by.y = "MGI.symbol",
                              all = FALSE
                              )


# Remove duplicated HGNC.symbol
tablecounts_tpm_plus <- tablecounts_tpm_plus[!duplicated(tablecounts_tpm_plus$HGNC.symbol),]

# Put symbol as rownames
rownames(tablecounts_tpm_plus) <- tablecounts_tpm_plus$HGNC.symbol
tablecounts_tpm_plus$HGNC.symbol <- NULL
tablecounts_tpm_plus$Row.names <- NULL


deconv_quantiseq_df <- deconvolute(gene_expression = tablecounts_tpm_plus, method=method_deconv) %>% column_to_rownames("cell_type") %>% as.data.frame()

colnames(deconv_quantiseq_df) <- sample_plan$new_name 

```

```{r}

deconvolute_MTX <- function(method_deconv, filename_1){
  
  deconv_quantiseq_df <- deconvolute(gene_expression = tablecounts_tpm_plus, method=method_deconv) %>% column_to_rownames("cell_type") %>% as.data.frame()
  
  colnames(deconv_quantiseq_df) <- sample_plan$new_name
  
  deconv_quantiseq_df_MP <- deconv_quantiseq_df %>% dplyr::select(sample_plan_MP$new_name)
  
  # Create annotations
  top_annot_MP <- HeatmapAnnotation(df = sample_plan_MP["Treatment"],
                               col = list(Treatment = c("MTX_D3" = "lightgreen", "PBS_D3" = "seagreen", "MTX_D10" = "black", "PBS_D10" = "red", "MTX_D22" = "yellow", "PBS_D22" = "purple")))
  
  # Heatmap 
  png(filename_1, height = 800, width = 1200)
  heatmap_MP_deconv_quantiseq <- as.matrix(deconv_quantiseq_df_MP[rownames(deconv_quantiseq_df_MP) != "uncharacterized cell",])
  pmap_MP_deconv_quantiseq <- ComplexHeatmap::Heatmap(heatmap_MP_deconv_quantiseq,
                                      name = "prMP_IC",
                                      cluster_columns = FALSE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = factor(sample_plan_MP$Treatment, levels = c('PBS_D3','MTX_D3', 'PBS_D10', 'MTX_D10', 'PBS_D22', 'MTX_D22')),
                                      cluster_rows = FALSE,
                                      row_names_gp = grid::gpar(fontsize = 13, fontface="bold"),
                                      top_annotation = top_annot_MP,
                                      show_column_names = FALSE,
                                      column_order = rownames(sample_plan_MP))


print(pmap_MP_deconv_quantiseq)
dev.off()
}



deconvolute_OCP <- function(method_deconv, filename_1){
  
  deconv_quantiseq_df <- deconvolute(gene_expression = tablecounts_tpm_plus, method=method_deconv) %>% column_to_rownames("cell_type") %>% as.data.frame()
  
  colnames(deconv_quantiseq_df) <- sample_plan$new_name
  
  deconv_quantiseq_df_OCP <- deconv_quantiseq_df %>% dplyr::select(sample_plan_OCP$new_name)
  
  top_annot_OCP <- HeatmapAnnotation(df = sample_plan_OCP["Treatment"], col = list(Treatment = c("Oxa_D3" = "lightgreen", "Oxa_D10" = "blue", "Oxa_D22" = "orange", "Cis_D3" = "purple", "Cis_D10" = "black", "Cis_D22" = "grey", "PBS_D3" = "seagreen", "PBS_D10" = "red", "PBS_D22" = "yellow")))
  
  # Heatmap 
  png(filename_1, height = 800, width = 1200)
  heatmap_OCP_deconv_quantiseq <- as.matrix(deconv_quantiseq_df_OCP[rownames(deconv_quantiseq_df_OCP) != "uncharacterized cell",])
  pmap_OCP_deconv_quantiseq <- ComplexHeatmap::Heatmap(heatmap_OCP_deconv_quantiseq,
                                      name = "prop_IC",
                                      cluster_columns = TRUE,
                                      cluster_column_slices = FALSE,
                                      show_column_dend = FALSE,
                                      column_split = factor(sample_plan_OCP$Treatment, levels = c('PBS_D3','Oxa_D3', 'Cis_D3', 'PBS_D10', 'Oxa_D10', 'Cis_D10', 'PBS_D22', 'Oxa_D22', 'Cis_D22')),
                                      cluster_rows = FALSE,
                                      #column_title = c(factor(levels(sample_plan_OCP$Treatment))),
                                      clustering_method_columns = "ward.D2", 
                                      row_names_gp = grid::gpar(fontsize = 13, fontface="bold"),
                                      top_annotation = top_annot_OCP,
                                      show_column_names = FALSE)


print(pmap_OCP_deconv_quantiseq)
dev.off()
}


```


### **5.1.1. MTX and PBS_MTX**

```{r}
deconvolute_MTX("quantiseq", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_quantiseq_MP.png")
deconvolute_MTX("mcp_counter", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_mcp_counter_MP.png")
deconvolute_MTX("epic", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_epic_MP.png")
deconvolute_MTX("cibersort", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_cibersort_MP.png")
```

### **5.1.2. Oxa/Cis and PBS_OCP**

```{r}
deconvolute_OCP("quantiseq", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_quantiseq_OCP.png")
deconvolute_OCP("mcp_counter", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_mcp_counter_OCP.png")
deconvolute_OCP("epic", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_epic_OCP.png")
deconvolute_OCP("cibersort", filename_1 = "/data/users/tlassale/MODICED_link/Bulk_RNA_in_vivo/in_vivo_results/Heatmap_deconv_cibersort_OCP.png")
```


